%% Generated by Sphinx.
\def\sphinxdocclass{article}
\documentclass[a4paper,10pt,english]{sphinxhowto}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{times}
\expandafter\ifx\csname T@LGR\endcsname\relax
\else
% LGR was declared as font encoding
  \substitutefont{LGR}{\rmdefault}{cmr}
  \substitutefont{LGR}{\sfdefault}{cmss}
  \substitutefont{LGR}{\ttdefault}{cmtt}
\fi
\expandafter\ifx\csname T@X2\endcsname\relax
  \expandafter\ifx\csname T@T2A\endcsname\relax
  \else
  % T2A was declared as font encoding
    \substitutefont{T2A}{\rmdefault}{cmr}
    \substitutefont{T2A}{\sfdefault}{cmss}
    \substitutefont{T2A}{\ttdefault}{cmtt}
  \fi
\else
% X2 was declared as font encoding
  \substitutefont{X2}{\rmdefault}{cmr}
  \substitutefont{X2}{\sfdefault}{cmss}
  \substitutefont{X2}{\ttdefault}{cmtt}
\fi


\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=\small}
\usepackage{geometry}

% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}
\addto\captionsenglish{\renewcommand{\contentsname}{Contents:}}

\usepackage{sphinxmessages}
\setcounter{tocdepth}{1}



\title{EpiGEN: an epistasis simulation pipeline}
\date{Sep 14, 2020}
\release{}
\author{David B. Blumenthal, Lorenzo Viola, Markus List, Jan Baumbach, Paolo Tieri, and Tim Kacprowski}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{}
\makeindex
\begin{document}

\pagestyle{empty}
\makeatletter\py@HeaderFamily\raggedright{\huge\@title}\\[24pt]{\Large User Guide}\\[24pt]{\large\@author}\makeatother\normalfont
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}



\section{Getting Started with EpiGEN}
\label{\detokenize{README:getting-started-with-epigen}}\label{\detokenize{README::doc}}

\subsection{Scope}
\label{\detokenize{README:scope}}
EpiGEN is an easy-to-use epistasis simulation pipeline written in Python. It supports epistasis models of arbitrary size, which can be specified either extensionally or via parametrized risk models. Moreover, the user can specify the minor allele frequencies (MAFs) of both noise and disease SNPs, and provide a biased target distribution for the generated phenotypes to simulate observation bias.


\subsection{Installation}
\label{\detokenize{README:installation}}
EpiGen is freely available on \sphinxhref{https://github.com/baumbachlab/epigen}{GitHub}. Before installing it on your machine, make sure that you have \sphinxhref{https://git-scm.com/}{git} and \sphinxhref{https://git-lfs.github.com/}{git lfs} installed. Then simply execute the following line in a terminal:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
git clone https://github.com/baumbachlab/epigen
\end{sphinxVerbatim}


\subsection{Usage}
\label{\detokenize{README:usage}}
The user interface of EpiGEN consists of three scripts:
\begin{itemize}
\item {} 
\sphinxcode{\sphinxupquote{simulate\_data.py}}

\item {} 
\sphinxcode{\sphinxupquote{generate\_genotype\_corpus.py}}

\item {} 
\sphinxcode{\sphinxupquote{merge\_genotype\_corpora.py}}

\end{itemize}

The script \sphinxcode{\sphinxupquote{simulate\_data.py}} simulates epistasis data on top of a pre-computed genotype corpus. For each chromosome \sphinxcode{\sphinxupquote{\textless{}CHROM\textgreater{}}} and each HAPMAP3 population code \sphinxcode{\sphinxupquote{\textless{}POP\textgreater{}}}, EpiGEN contains a pre-computed corpus for 10000 individuals, which is identified by the prefix \sphinxcode{\sphinxupquote{corpora/\textless{}CHROM\textgreater{}\_\textless{}POP\textgreater{}\_}}. For example, if you want to generate epistasis data with ID 0 for 7500 individuals and 10000 SNPs on top of the pre-computed corpus \sphinxcode{\sphinxupquote{corpora/1\_ASW\_}}, where the parametrized epistasis model \sphinxcode{\sphinxupquote{models/param\_model.xml}} acts upon the SNPs with IDs 156, 3, and 1076 in the corpus, you can use \sphinxcode{\sphinxupquote{simulate\_data.py}} as follows:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
python3 simulate\PYGZus{}data.py \PYGZhy{}\PYGZhy{}sim\PYGZhy{}ids \PYG{l+m}{0} \PYGZhy{}\PYGZhy{}corpus\PYGZhy{}id \PYG{l+m}{1} \PYGZhy{}\PYGZhy{}pop ASW \PYGZhy{}\PYGZhy{}inds \PYG{l+m}{7500} \PYGZhy{}\PYGZhy{}snps \PYG{l+m}{10000} \PYGZhy{}\PYGZhy{}disease\PYGZhy{}snps \PYG{l+m}{156} \PYG{l+m}{3} \PYG{l+m}{1076} \PYGZhy{}\PYGZhy{}model models/param\PYGZus{}model.xml  
\end{sphinxVerbatim}

As you will notice when executing this command, a large fraction of the runtime of \sphinxcode{\sphinxupquote{simulate\_data.py}} is used for loading the corpora. If you want to simulate data for only a small number of individuals, it is therefore advisable to first compute your own, smaller corpora. You can also speed-up the script by unzipping the corpora before running it.

If you want to use custom corpora instead of the pre-computed ones, you can generate them via the script \sphinxcode{\sphinxupquote{generate\_genotype\_corpus.py}}. For example, the corpus \sphinxcode{\sphinxupquote{corpora/1\_ASW\_}} shipped with EpiGEN was generated as follows:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
python3 generate\PYGZus{}genotype\PYGZus{}corpus.py \PYGZhy{}\PYGZhy{}corpus\PYGZhy{}id \PYG{l+m}{1} \PYGZhy{}\PYGZhy{}pop ASW \PYGZhy{}\PYGZhy{}inds \PYG{l+m}{10000} \PYGZhy{}\PYGZhy{}chroms \PYG{l+m}{1} \PYGZhy{}\PYGZhy{}compress 
\end{sphinxVerbatim}

Finally, the script \sphinxcode{\sphinxupquote{merge\_genotype\_corpora.py}} allows you to merge pre-computed corpora into a larger copus. For instance, the following command merges the pre-computed corpora \sphinxcode{\sphinxupquote{corpora/1\_ASW\_}} and \sphinxcode{\sphinxupquote{corpora/2\_ASW\_}} into a newly generated corpus \sphinxcode{\sphinxupquote{corpora/23\_ASW\_}}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
python3 merge\PYGZus{}genotype\PYGZus{}corpora.py \PYGZhy{}\PYGZhy{}corpus\PYGZhy{}ids \PYG{l+m}{1} \PYG{l+m}{2} \PYGZhy{}\PYGZhy{}pops ASW ASW \PYGZhy{}\PYGZhy{}corpus\PYGZhy{}id \PYG{l+m}{23} \PYGZhy{}\PYGZhy{}append SNPS
\end{sphinxVerbatim}

Detailed descriptions of how to use the scripts can be found in the HTML and PDF documentations contained in \sphinxcode{\sphinxupquote{docs/build/html}} and \sphinxcode{\sphinxupquote{docs/build/latex}}.


\subsection{Implementing Custom Interaction Models}
\label{\detokenize{README:implementing-custom-interaction-models}}
EpiGEN natively supports four parametrized interaction models: exponential, multiplicative, joint-dominant, and joint-recessive interaction. Further interaction models can easily be implemented by the user. Assume, for instance, that the user wants to implement xor-dominant interaction, i.e., a parametrized interaction model where there is an effect if and only if there is at least one minor allele at exactly one of the SNPs involved in the interaction. Then it suffices to insert the following five lines of code at line 242 of \sphinxcode{\sphinxupquote{utils/parametrized\_model.py}}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k}{elif} \PYG{n}{model\PYGZus{}type} \PYG{o}{==} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{xor\PYGZhy{}dominant}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}
	\PYG{k}{if} \PYG{n}{np}\PYG{o}{.}\PYG{n}{sum}\PYG{p}{(}\PYG{n}{gen\PYGZus{}at\PYGZus{}snp\PYGZus{}set}\PYG{p}{[}\PYG{n}{poss}\PYG{p}{]}\PYG{p}{)} \PYG{o}{==} \PYG{l+m+mi}{1}\PYG{p}{:}
		\PYG{k}{return} \PYG{n}{alpha}
	\PYG{k}{else}\PYG{p}{:}
		\PYG{k}{return} \PYG{l+m+mi}{1}
\end{sphinxVerbatim}

For consistency, it is also recommendable to add the string \sphinxcode{\sphinxupquote{"xor-dominant"}} to the error message on line 249 of \sphinxcode{\sphinxupquote{utils/parametrized\_model.py}}, as well to the list of acceptable interaction types on line 42 of the document type definition \sphinxcode{\sphinxupquote{models/ParametrizedModel.dtd}}.


\subsection{Requirements}
\label{\detokenize{README:requirements}}
EpiGEN has the following dependencies:
\begin{itemize}
\item {} 
Python 3.3 or higher.

\item {} 
Numpy 1.17.3 or higher.

\item {} 
Scipy 1.3.1 or higher.

\item {} 
Matplotlib 3.1.1 or higher.

\end{itemize}

Moreover, due to its HAPGEN2 dependency, the script \sphinxcode{\sphinxupquote{generate\_genotype\_corpus.py}} needs to be run on a Linux machine or on a machine running macOS 10.14 or lower. However, you can avoid running \sphinxcode{\sphinxupquote{generate\_genotype\_corpus.py}} by using the pre-computed corpora and merging them, if necessary.

If you want to re-compile the documentation contained in the \sphinxcode{\sphinxupquote{docs}} directory, you additionally need to install Sphinx, the extension recommonmark, and the package mock. If you have these packages installed, the HTML and PDF documentations can be re-compiled by executing \sphinxcode{\sphinxupquote{make html}} and \sphinxcode{\sphinxupquote{make latexpdf}} from the \sphinxcode{\sphinxupquote{docs}} directory.


\subsection{License}
\label{\detokenize{README:license}}
All of EpiGEN’s Python sources are licensed under the \sphinxhref{https://www.gnu.org/licenses/gpl-3.0.de.html}{GNU General Public License 3}. However, this license does not cover the HAPGEN2 binaries, which are distributed with EpiGEN and are called by the script \sphinxcode{\sphinxupquote{generate\_genotype\_corpus.py}}. HAPGEN2 is property of the University of Oxford and may only be freely used for academic research and in accordance with the license found at \sphinxhref{https://mathgen.stats.ox.ac.uk/genetics\_software/hapgen/LICENCE}{https://mathgen.stats.ox.ac.uk/genetics\_software/hapgen/LICENCE}. Copies of the GNU General Public License 3 and of the license for HAPGEN2 are distributed with EpiGEN.


\subsection{Citing EpiGEN}
\label{\detokenize{README:citing-epigen}}
If you use EpiGEN, please cite the following paper:
\begin{itemize}
\item {} 
D. B. Blumenthal, L. Viola, M. List, J. Baumbach, P. Tieri, T. Kacprowski (2020). “EpiGEN: an epistasis simulation pipeline”, Bioinformatics, DOI: \sphinxhref{https://doi.org/10.1093/bioinformatics/btaa245}{10.1093/bioinformatics/btaa245}.

\end{itemize}


\subsection{Structure of the Repository}
\label{\detokenize{README:structure-of-the-repository}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
.
├── README.md                        // README
├── LICENSE                          // A copy of the GNU General Public License 3
├── requirements.txt                 // Lists dependencies
├── simulate\PYGZus{}data.py                 // Script to simulate epistasis data
├── generate\PYGZus{}genotype\PYGZus{}corpus.py      // Script to generate genotype corpus
├── merge\PYGZus{}genotype\PYGZus{}corpora.py        // Script to merge genotype corpora
├── validate\PYGZus{}simulated\PYGZus{}data.py       // Script to validate simulated data
├── test\PYGZus{}runtime.py                  // Script to test EpiGEN\PYGZsq{}s runtime performance
├── docs                             // Contains Sphinx documentation
├── sim                              // Output directory for simulated data
├── corpora                          // Output directory for genotype corpora
├── temp                             // Contains auxiliary files 
├── ext                              // Contains external libraries and data
│   ├── HAPGEN2                      // Contains HAPGEN2 binaries and license
│   └── HAPMAP3                      // Contains HAPMAP3 data
├── models                           // Contains epistasis models
│   ├── ParametrizedModel.dtd        // Doctype definition for parametrized models
│   ├── ext\PYGZus{}model.ini                // An example of an extensional model
│   ├── param\PYGZus{}model.xml              // An example of a parametrized model
│   └── ...                          // Further models
└── utils                            // Contains the core of EpiGEN
    ├── \PYGZus{}\PYGZus{}init\PYGZus{}\PYGZus{}.py                  // \PYGZus{}\PYGZus{}init\PYGZus{}\PYGZus{} file
    ├── data\PYGZus{}simulator.py            // Implements simulation of epistasis data
    ├── genotype\PYGZus{}corpus\PYGZus{}generator.py // Implements generation of genotype corpora
    ├── genotype\PYGZus{}corpusmerger.py     // Implements merging of genotype corpora
    ├── parametrized\PYGZus{}model.py.       // Implements parametrized models 
    ├── extensional\PYGZus{}model.py.        // Implements extensional models
    └── argparse\PYGZus{}checks.py.          // Implements argparse checks
\end{sphinxVerbatim}


\section{The script \sphinxstyleliteralintitle{\sphinxupquote{simulate\_data.py}}}
\label{\detokenize{simulate_data:module-simulate_data}}\label{\detokenize{simulate_data:the-script-simulate-data-py}}\label{\detokenize{simulate_data::doc}}\index{simulate\_data (module)@\spxentry{simulate\_data}\spxextra{module}}
This script constitutes the main user interface of EpiGEN \textendash{} run it, to simulate epistasis data.

This script has to be run on top of a pre-computed genotype corpus. For each population code \sphinxcode{\sphinxupquote{\textless{}POP\textgreater{}}}
and each chromosome \sphinxcode{\sphinxupquote{\textless{}CHROM\textgreater{}}}, EpiGEN contains a pre-computed corpus for 10000 individuals. These corpora
can be selected by running the script with the options \sphinxcode{\sphinxupquote{-{-}corpus-id \textless{}CRHOM\textgreater{} -{-}pop \textless{}POP\textgreater{}}}.
You can also use your own corpora \textendash{} simply run the script \sphinxcode{\sphinxupquote{generate\_genotype\_corpus.py}} before running this script.

\sphinxstylestrong{Usage}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{python3} \PYG{n}{simulate\PYGZus{}data}\PYG{o}{.}\PYG{n}{py} \PYG{p}{[}\PYG{n}{required} \PYG{n}{arguments}\PYG{p}{]} \PYG{p}{[}\PYG{n}{optional} \PYG{n}{arguments}\PYG{p}{]}
\end{sphinxVerbatim}
\begin{description}
\item[{\sphinxstylestrong{Required Arguments:}}] \leavevmode\begin{description}
\item[{\sphinxcode{\sphinxupquote{-{-}corpus-id CORPUS\_ID}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
ID of selected genotype corpus.

\item[{Accepted Arguments:}] \leavevmode
Non-negative integers.

\item[{Effect:}] \leavevmode
Together with \sphinxcode{\sphinxupquote{-{-}pop}}, this option selects the genotype corpus with the prefix \sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}}}.
If this corpus does not exist, the script raises an error. If necessary, run the script \sphinxcode{\sphinxupquote{./generate\_genotype\_corpus.py}}
to generate the desired corpus.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-{-}pop POP}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
HAPMAP3 population code of selected genotype corpus.

\item[{Accepted Arguments:}] \leavevmode
ASW, CEU, CEU+TSI, CHD, GIH, JPT+CHB, LWK, MEX, MKK, TSI, and MIX (for merged corpora).

\item[{Effect:}] \leavevmode
Together with \sphinxcode{\sphinxupquote{-{-}corpus-id}}, this option selects the genotype corpus with the prefix \sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}}}.
If this corpus does not exist, the script raises an error. If necessary, run the script \sphinxcode{\sphinxupquote{./generate\_genotype\_corpus.py}}
to generate the desired corpus.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-{-}model MODEL}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
Path to epistasis model given as INI or XML file.
INI files are used to provide extensionally defined models, XML files specify parametrized models.

\item[{Accepted Arguments:}] \leavevmode
Strings ending in \sphinxcode{\sphinxupquote{.ini}} or \sphinxcode{\sphinxupquote{.xml}} that represent paths to existing model files.

\item[{Effect:}] \leavevmode
Specifies the epistasis model used by the simulator.

\item[{Format of INI files for extensionally defined models:}] \leavevmode
For each genotype of length \sphinxcode{\sphinxupquote{\textless{}size\textgreater{}}}, parameters of a Normal distribution must be provided.
INI files for quantitative phenotypes have to be of the following format::

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{p}{[}\PYG{n}{Model} \PYG{n}{Type}\PYG{p}{]}
\PYG{n}{size} \PYG{o}{=} \PYG{o}{\PYGZlt{}}\PYG{n}{size}\PYG{o}{\PYGZgt{}}
\PYG{n}{phenotype} \PYG{o}{=} \PYG{n}{quantitative}
\PYG{p}{[}\PYG{n}{Model} \PYG{n}{Definition}\PYG{p}{]}
\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{o}{.}\PYG{o}{.}\PYG{o}{.}\PYG{p}{,}\PYG{l+m+mi}{0} \PYG{o}{=} \PYG{o}{\PYGZlt{}}\PYG{n}{mu}\PYG{o}{\PYGZgt{}}\PYG{p}{,}\PYG{o}{\PYGZlt{}}\PYG{n}{stdev}\PYG{o}{\PYGZgt{}}
\PYG{o}{.}
\PYG{o}{.}
\PYG{o}{.}
\PYG{l+m+mi}{2}\PYG{p}{,}\PYG{o}{.}\PYG{o}{.}\PYG{o}{.}\PYG{p}{,}\PYG{l+m+mi}{2} \PYG{o}{=} \PYG{o}{\PYGZlt{}}\PYG{n}{mu}\PYG{o}{\PYGZgt{}}\PYG{p}{,}\PYG{o}{\PYGZlt{}}\PYG{n}{stddev}\PYG{o}{\PYGZgt{}}
\end{sphinxVerbatim}

For each genotype of length \sphinxcode{\sphinxupquote{\textless{}size\textgreater{}}}, parameters of a categorical distribution must be provided
INI files for categorical phenotypes have to be of the following format::

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{p}{[}\PYG{n}{Model} \PYG{n}{Type}\PYG{p}{]}
\PYG{n}{size} \PYG{o}{=} \PYG{o}{\PYGZlt{}}\PYG{n}{size}\PYG{o}{\PYGZgt{}}
\PYG{n}{phenotype} \PYG{o}{=} \PYG{o}{\PYGZlt{}}\PYG{n}{c}\PYG{o}{\PYGZgt{}}
\PYG{p}{[}\PYG{n}{Model} \PYG{n}{Definition}\PYG{p}{]}
\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{o}{.}\PYG{o}{.}\PYG{o}{.}\PYG{p}{,}\PYG{l+m+mi}{0} \PYG{o}{=} \PYG{o}{\PYGZlt{}}\PYG{n}{p\PYGZus{}1}\PYG{o}{\PYGZgt{}}\PYG{p}{,}\PYG{o}{.}\PYG{o}{.}\PYG{o}{.}\PYG{p}{,}\PYG{o}{\PYGZlt{}}\PYG{n}{p\PYGZus{}c}\PYG{o}{\PYGZgt{}}
\PYG{o}{.}
\PYG{o}{.}
\PYG{o}{.}
\PYG{l+m+mi}{2}\PYG{p}{,}\PYG{o}{.}\PYG{o}{.}\PYG{o}{.}\PYG{p}{,}\PYG{l+m+mi}{2} \PYG{o}{=} \PYG{o}{\PYGZlt{}}\PYG{n}{p\PYGZus{}1}\PYG{o}{\PYGZgt{}}\PYG{p}{,}\PYG{o}{.}\PYG{o}{.}\PYG{o}{.}\PYG{p}{,}\PYG{o}{\PYGZlt{}}\PYG{n}{p\PYGZus{}c}\PYG{o}{\PYGZgt{}}
\end{sphinxVerbatim}

\item[{Format of XML files for parametrized models:}] \leavevmode
XML files have to match the document type definition \sphinxcode{\sphinxupquote{models/ParametrizedModel.dtd}}.

\item[{Examples:}] \leavevmode
Cf. the files \sphinxcode{\sphinxupquote{models/ext\_model.ini}} and \sphinxcode{\sphinxupquote{models/param\_model.xml}}.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-{-}snps SNPS}}:}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
Number of SNPs in simulated data.

\item[{Accepted Arguments:}] \leavevmode
Positive integers. If larger than the number of SNPs in the selected corpus, it is lowered to this number.
Should be set to a number that is significantly smaller than the number of SNPs in the selected corpus,
because otherwise, EpiGEN’s subsampling techniques have no effect.

\item[{Effect:}] \leavevmode
Determines how many SNPs from the selected corpus are included in the simulated data.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-{-}inds INDS}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
Number of individuals in simulated data.

\item[{Accepted Arguments:}] \leavevmode
Positive integers. If larger than the number of individuals in the selected corpus, it is lowered to this number.
Should be set to a number that is significantly smaller than the number of individuals in the selected corpus,
because otherwise, EpiGEN’s subsampling techniques have no effect.

\item[{Effect:}] \leavevmode
Determines how many individuals from the selected corpus are included in the simulated data.

\end{description}

\end{description}

\item[{\sphinxstylestrong{Required Group of Mutually Exclusive Arguments:}}] \leavevmode\begin{description}
\item[{\sphinxcode{\sphinxupquote{-{-}sim-ids SIM\_ID {[}SIM\_ID ...{]}}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
IDs of simulated data.

\item[{Accepted Arguments:}] \leavevmode
One or several non-negative integers.

\item[{Effect:}] \leavevmode
If N IDs \textless{}SIM\_ID\_1\textgreater{} … \textless{}SIM\_ID\_N\textgreater{} are provided, N simulated epistasis instances with prefices
\sphinxcode{\sphinxupquote{./sim/\textless{}SIM\_ID\_1\textgreater{}\_\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}}}, …, \sphinxcode{\sphinxupquote{./sim/\textless{}SIM\_ID\_N\textgreater{}\_\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}}} are generated.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-{-}num-sims NUM\_SIMS}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
The number of epistasis instances that should be generated.

\item[{Accepted Arguments:}] \leavevmode
Positive integer.

\item[{Effect:}] \leavevmode
If specified, \textless{}NUM\_SIMS\textgreater{} simulated epistasis instances with prefices
\sphinxcode{\sphinxupquote{./sim/0\_\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}}}, …, \sphinxcode{\sphinxupquote{./sim/\textless{}NUM\_SIMS-1\textgreater{}\_\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}}} are generated.

\end{description}

\end{description}

\item[{\sphinxstylestrong{Optional Arguments:}}] \leavevmode\begin{description}
\item[{\sphinxcode{\sphinxupquote{-{-}global-maf-range LB UB}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
Range of acceptable MAFs for noise SNPs.

\item[{Accepted Arguments:}] \leavevmode
Floats \sphinxcode{\sphinxupquote{\textless{}LB\textgreater{}}} and \sphinxcode{\sphinxupquote{\textless{}UB\textgreater{}}} with \sphinxcode{\sphinxupquote{0 \textless{}= \textless{}LB\textgreater{} \textless{} \textless{}UB\textgreater{} \textless{}= 1}}.

\item[{Default:}] \leavevmode
{[}0,1{]}

\item[{Effect:}] \leavevmode
All SNPs except for the disease SNPs are randomly sampled from those SNPs in the corpus whose MAFs
fall into the specified range. If the range is too narrow, it is dynamically extended at runtime.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-{-}biased-distr PARAM {[}PARAM ...{]}}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
Biased target distribution for simulated phenotypes.

\item[{Accepted Arguments for Quantitative Phenotypes:}] \leavevmode
A white-space separated list of floats of length 2 whose elements represent the mean (first element) and
standard deviation (second element) of a Normal distribution.

\item[{Accepted Arguments for Categorical Phenotypes:}] \leavevmode
A white-space separated list of floats of length \sphinxcode{\sphinxupquote{\textless{}c\textgreater{}}} whose entries represent the probabilities of the \sphinxcode{\sphinxupquote{\textless{}c\textgreater{}}}
categories.

\item[{Effect:}] \leavevmode
If provided, the individuals are subsampled after generating the phenotypes such that the obtained phenotype distribution
matches the biased distribution. This option can hence be used to model observation bias.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-{-}seed SEED}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
Seed for numpy.random.

\item[{Accepted Arguments:}] \leavevmode
Non-negative integers.

\item[{Effect:}] \leavevmode
If provided, the simulator always generates the same data given the same input.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-{-}compress}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
Compress the generated output files.

\item[{Accepted Arguments:}] \leavevmode
None.

\item[{Effect:}] \leavevmode
Determines the suffix \sphinxcode{\sphinxupquote{\textless{}SUFFIX\textgreater{}}} of the generated files. If provided, \sphinxcode{\sphinxupquote{\textless{}SUFFIX\textgreater{}}} is set to \sphinxcode{\sphinxupquote{json.bz2}}.
Otherwise, it is set to \sphinxcode{\sphinxupquote{json}}.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-h, -{-}help}}}] \leavevmode\begin{description}
\item[{Effect:}] \leavevmode
Show help message and exit.

\end{description}

\end{description}

\item[{\sphinxstylestrong{Optional Mutually Exclusive Arguments:}}] \leavevmode\begin{description}
\item[{\sphinxcode{\sphinxupquote{-{-}disease-snps SNP {[}SNP ...{]}}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
Position of disease SNPs in selected genotype corpus.

\item[{Accepted Arguments:}] \leavevmode
White space separated list of non-negative integers whose length matches the size of the model specified
via the option \sphinxcode{\sphinxupquote{-{-}model}}. All integers must be smaller than the number of SNPs in the selected corpus.

\item[{Effect:}] \leavevmode
If provided, the selected SNPs form the disease SNP set employed by the simulator.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-{-}disease-maf-range LB UB}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
Range of acceptable MAFs for disease SNPs.

\item[{Accepted Arguments:}] \leavevmode
Floats \sphinxcode{\sphinxupquote{\textless{}LB\textgreater{}}} and \sphinxcode{\sphinxupquote{\textless{}UB\textgreater{}}} with \sphinxcode{\sphinxupquote{0 \textless{}= \textless{}LB\textgreater{} \textless{} \textless{}UB\textgreater{} \textless{}= 1}}.

\item[{Default:}] \leavevmode
{[}0.1,1{]}

\item[{Effect:}] \leavevmode
Unless \sphinxcode{\sphinxupquote{-{-}disease-snps}} is provided, the disease SNPs are randomly sampled from those SNPs in the corpus whose MAFs
fall into the specified range. If the range is too narrow, it is dynamically extended at runtime.

\end{description}

\end{description}

\item[{\sphinxstylestrong{Output:}}] \leavevmode\begin{description}
\item[{\sphinxstyleemphasis{Output File:}}] \leavevmode\begin{description}
\item[{Path:}] \leavevmode
\sphinxcode{\sphinxupquote{./sim/\textless{}ID\textgreater{}\_\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}.\textless{}SUFFIX\textgreater{}}}

\item[{Format for Quantitative Phenotypes:}] \leavevmode
(Compressed) JSON file of the form \sphinxcode{\sphinxupquote{\{"num\_snps": \textless{}NUM\_SNPS\textgreater{}, "num\_inds": \textless{}NUM\_INDS\textgreater{}, "model\_type": "quantitative", "genotype": \textless{}GENOTYPE\_DATA\textgreater{}, "phenotype": \textless{}PHENOTYPE\_DATA\textgreater{}, "snps": \textless{}SNP\_DATA\textgreater{}, "disease\_snps": \textless{}DISEASE\_SNP\_DATA\textgreater{}, "mafs": \textless{}MAF\_DATA\textgreater{}\}}}.

\item[{Format for Categorical Phenotypes:}] \leavevmode
(Compressed) JSON file of the form \sphinxcode{\sphinxupquote{\{"num\_snps": \textless{}NUM\_SNPS\textgreater{}, "num\_inds": \textless{}NUM\_INDS\textgreater{}, "model\_type": "categorical", "num\_categories": \textless{}NUM\_CATEGORIES\textgreater{}, "genotype": \textless{}GENOTYPE\_DATA\textgreater{}, "phenotype": \textless{}PHENOTYPE\_DATA\textgreater{}, "snps": \textless{}SNP\_DATA\textgreater{}, "disease\_snps": \textless{}DISEASE\_SNP\_DATA\textgreater{}, "mafs": \textless{}MAF\_DATA\textgreater{}\}}}.

\end{description}

\item[{\sphinxcode{\sphinxupquote{\textless{}NUM\_SNPS\textgreater{}}}}] \leavevmode\begin{description}
\item[{Key:}] \leavevmode
\sphinxcode{\sphinxupquote{"num\_spns"}}

\item[{Content and Format:}] \leavevmode
Integer representing the number of SNPs.

\end{description}

\item[{\sphinxcode{\sphinxupquote{\textless{}NUM\_INDS\textgreater{}}}}] \leavevmode\begin{description}
\item[{Key:}] \leavevmode
\sphinxcode{\sphinxupquote{"num\_inds"}}

\item[{Content and Format:}] \leavevmode
Integer representing the number of individuals.

\end{description}

\item[{\sphinxcode{\sphinxupquote{\textless{}NUM\_CATEGORIES\textgreater{}}}}] \leavevmode\begin{description}
\item[{Key:}] \leavevmode
\sphinxcode{\sphinxupquote{"num\_categories"}}

\item[{Content and Format:}] \leavevmode
Integer representing the number of categories for categorical phenotypes.

\end{description}

\item[{\sphinxcode{\sphinxupquote{\textless{}GENOTYPE\_DATA\textgreater{}}}}] \leavevmode\begin{description}
\item[{Key:}] \leavevmode
\sphinxcode{\sphinxupquote{"genotype"}}

\item[{Content and Format:}] \leavevmode
JSON field of the form \sphinxcode{\sphinxupquote{{[}{[}G\_0\_0, ..., G\_0\_\textless{}INDS-1\textgreater{}{]} ... {[}G\_\textless{}SNPS-1\textgreater{}\_0, ..., G\_\textless{}SNPS-1\textgreater{}\_\textless{}INDS-1\textgreater{}{]}{]}}},
where \sphinxcode{\sphinxupquote{G\_S\_I}} encodes the number of minor alleles of the individual with index \sphinxcode{\sphinxupquote{I}} at the SNP with index \sphinxcode{\sphinxupquote{S}}.

\end{description}

\item[{\sphinxcode{\sphinxupquote{\textless{}PHENOTYPE\_DATA\textgreater{}}}}] \leavevmode\begin{description}
\item[{Key:}] \leavevmode
\sphinxcode{\sphinxupquote{"phenotype"}}

\item[{Content and Format:}] \leavevmode
JSON field of the form \sphinxcode{\sphinxupquote{{[}P\_0, ..., P\_\textless{}INDS-1\textgreater{}{]}}}, where \sphinxcode{\sphinxupquote{P\_I}} encodes the phenotype of
the individual with index \sphinxcode{\sphinxupquote{I}}.

\end{description}

\item[{\sphinxcode{\sphinxupquote{\textless{}SNP\_DATA\textgreater{}}}}] \leavevmode\begin{description}
\item[{Key:}] \leavevmode
\sphinxcode{\sphinxupquote{"snps"}}

\item[{Content and Format:}] \leavevmode
JSON field of the form \sphinxcode{\sphinxupquote{{[}INFO\_0, ..., INFO\_\textless{}SNPS-1\textgreater{}{]}}}, where \sphinxcode{\sphinxupquote{INFO\_S}} contains the following
information about the SNP with index \sphinxcode{\sphinxupquote{S}}: RS identifier, chromosome number, position on chromosome, major allele, minor allele.

\end{description}

\item[{\sphinxcode{\sphinxupquote{\textless{}DISEASE\_SNP\_DATA\textgreater{}}}}] \leavevmode\begin{description}
\item[{Key:}] \leavevmode
\sphinxcode{\sphinxupquote{"disease\_snps"}}

\item[{Content and Format:}] \leavevmode
JSON field of the form \sphinxcode{\sphinxupquote{{[}S\_0, ..., S\_\textless{}MODELSIZE-1\textgreater{}{]}}}, where \sphinxcode{\sphinxupquote{S\_POS}} encodes the index of the SNP at position \sphinxcode{\sphinxupquote{POS}}
in the epistasis model.

\end{description}

\item[{\sphinxcode{\sphinxupquote{\textless{}MAF\_DATA\textgreater{}}}}] \leavevmode\begin{description}
\item[{Key:}] \leavevmode
\sphinxcode{\sphinxupquote{"mafs"}}

\item[{Content and Format:}] \leavevmode
JSON field of the form \sphinxcode{\sphinxupquote{{[}MAF\_0, ..., MAF\_\textless{}SNPS-1\textgreater{}{]}}}, where \sphinxcode{\sphinxupquote{MAF\_S}} encodes the MAF of
the SNP with index \sphinxcode{\sphinxupquote{S}}.

\end{description}

\end{description}

\end{description}
\index{run\_script() (in module simulate\_data)@\spxentry{run\_script()}\spxextra{in module simulate\_data}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{simulate_data:simulate_data.run_script}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{simulate\_data.}}\sphinxbfcode{\sphinxupquote{run\_script}}}{}{}
Runs the script.

\end{fulllineitems}



\section{The script \sphinxstyleliteralintitle{\sphinxupquote{generate\_genotype\_corpus.py}}}
\label{\detokenize{generate_genotype_corpus:module-generate_genotype_corpus}}\label{\detokenize{generate_genotype_corpus:the-script-generate-genotype-corpus-py}}\label{\detokenize{generate_genotype_corpus::doc}}\index{generate\_genotype\_corpus (module)@\spxentry{generate\_genotype\_corpus}\spxextra{module}}
Run this script to generate genotype corpora.

For each HAPMAP3 population and each chromosome, EpiGEN comes with a pre-computed corpus for 10000 individuals.
The corpora for chromosome \textless{}CHROM\textgreater{} have corpus ID \textless{}CHROM\textgreater{} and can be used by \sphinxcode{\sphinxupquote{simulate\_data.py}} right away.
If you want to simulate data on top of your own corpora, run this script.

\sphinxstylestrong{Usage}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{python3} \PYG{n}{generate\PYGZus{}genotype\PYGZus{}corpus}\PYG{o}{.}\PYG{n}{py} \PYG{p}{[}\PYG{n}{required} \PYG{n}{arguments}\PYG{p}{]} \PYG{p}{[}\PYG{n}{optional} \PYG{n}{arguments}\PYG{p}{]}
\end{sphinxVerbatim}
\begin{description}
\item[{\sphinxstylestrong{Requried Arguments:}}] \leavevmode\begin{description}
\item[{\sphinxcode{\sphinxupquote{-{-}corpus-id CORPUS\_ID}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
ID of generated genotype corpus.

\item[{Accepted Arguments:}] \leavevmode
Non-negative integers. If contained in range(1,23), the pre-computed corpora shipped with EpiGEN are overwritten.

\item[{Effect:}] \leavevmode
Together with \sphinxcode{\sphinxupquote{-{-}pop}}, this option determines the prefix \sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}}} of the files
that contain the generated corpus.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-{-}pop POP}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
HAPMAP3 population code of generated genotype corpus.

\item[{Accepted Arguments:}] \leavevmode
ASW, CEU, CEU+TSI, CHD, GIH, JPT+CHB, LWK, MEX, MKK, and TSI.

\item[{Effect:}] \leavevmode
Together with \sphinxcode{\sphinxupquote{-{-}pop}}, this option determines the prefix \sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}}} of the files
that contain the generated corpus.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-{-}inds INDS}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
Number of individuals in the corpus.

\item[{Accepted Arguments:}] \leavevmode
Positive integers.

\item[{Effect:}] \leavevmode
Determines how many individuals are contained in the generated corpus.

\end{description}

\end{description}

\item[{\sphinxstylestrong{Optional Arguments:}}] \leavevmode\begin{description}
\item[{\sphinxcode{\sphinxupquote{-{-}chroms CHROM {[}CHROM ...{]}}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
List of chromosomes for which the corpus should be generated.

\item[{Accepted Arguments:}] \leavevmode
A white-space separated list of integers between 1 and 22.

\item[{Default:}] \leavevmode
{[}1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22{]}

\item[{Effect:}] \leavevmode
For each selected chromosome, a corpus is generated by calling HAPGEN2 without disease SNPs.
The final corpus is then obtained by concatenating the corpora.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-{-}compress}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
Compress the generated output files.

\item[{Accepted Arguments:}] \leavevmode
None.

\item[{Effect:}] \leavevmode
Determines the suffix \sphinxcode{\sphinxupquote{\textless{}SUFFIX\textgreater{}}} of the generated files. If provided, \sphinxcode{\sphinxupquote{\textless{}SUFFIX\textgreater{}}} is set to \sphinxcode{\sphinxupquote{json.bz2}}.
Otherwise, it is set to \sphinxcode{\sphinxupquote{json}}.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-h, -{-}help}}}] \leavevmode\begin{description}
\item[{Effect:}] \leavevmode
Show help message and exit.

\end{description}

\end{description}

\item[{\sphinxstylestrong{Output:}}] \leavevmode\begin{description}
\item[{\sphinxstyleemphasis{Genotype Data:}}] \leavevmode\begin{description}
\item[{File:}] \leavevmode
\sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}\_genotype.\textless{}SUFFIX\textgreater{}}}

\item[{Content and Format:}] \leavevmode
Compressed JSON file of the form \sphinxcode{\sphinxupquote{{[}{[}G\_0\_0, ..., G\_0\_\textless{}INDS-1\textgreater{}{]}, ..., {[}G\_\textless{}SNPS-1\textgreater{}\_0, ..., G\_\textless{}SNPS-1\textgreater{}\_\textless{}INDS-1\textgreater{}{]}{]}}},
where \sphinxcode{\sphinxupquote{G\_S\_I}} encodes the number of minor alleles of the individual with index \sphinxcode{\sphinxupquote{I}} at the SNP with index \sphinxcode{\sphinxupquote{S}}.

\end{description}

\item[{\sphinxstyleemphasis{SNPs:}}] \leavevmode\begin{description}
\item[{File:}] \leavevmode
\sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}\_snps.\textless{}SUFFIX\textgreater{}}}

\item[{Content and Format:}] \leavevmode
Compressed JSON file of the form \sphinxcode{\sphinxupquote{{[}INFO\_0, ..., INFO\_\textless{}SNPS-1\textgreater{}{]}}}, where \sphinxcode{\sphinxupquote{INFO\_S}} contains the following
information about the SNP with index \sphinxcode{\sphinxupquote{S}}: RS identifier, chromosome number, position on chromosome, major allele, minor allele.

\end{description}

\item[{\sphinxstyleemphasis{MAFs:}}] \leavevmode\begin{description}
\item[{File:}] \leavevmode
\sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}\_mafs.\textless{}SUFFIX\textgreater{}}}

\item[{Content and Format:}] \leavevmode
Compressed JSON file of the form \sphinxcode{\sphinxupquote{{[}MAF\_0, ..., MAF\_\textless{}SNPS-1\textgreater{}{]}}}, where \sphinxcode{\sphinxupquote{MAF\_S}} encodes the MAF of
the SNP with index \sphinxcode{\sphinxupquote{S}}.

\end{description}

\item[{\sphinxstyleemphasis{Cumulative MAF distribution:}}] \leavevmode\begin{description}
\item[{File:}] \leavevmode
\sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}\_cum\_mafs.\textless{}SUFFIX\textgreater{}}}

\item[{Content and Format:}] \leavevmode
Compressed ordered JSON file of the form \sphinxcode{\sphinxupquote{{[}{[}MAF\_0, COUNT\_0{]}, ..., {[}MAF\_\textless{}NMAFS-1\textgreater{}, COUNT\_\textless{}NMAFS-1\textgreater{}{]}{]}}}, where \sphinxcode{\sphinxupquote{COUNT\_POS}} encodes the
number of SNPs is does not exceed \sphinxcode{\sphinxupquote{MAF\_POS}}.

\end{description}

\item[{\sphinxstyleemphasis{Plot of Cumulative MAF distribution:}}] \leavevmode\begin{description}
\item[{File:}] \leavevmode
\sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}\_cum\_mafs.pdf}}

\item[{Content and Format:}] \leavevmode
Plot of cumulative MAF distribution. Can be used to determine feasible ranges of MAFs passed to the options
\sphinxcode{\sphinxupquote{-{-}global-maf-range}} and \sphinxcode{\sphinxupquote{-{-}disease-maf-range}} of the script \sphinxcode{\sphinxupquote{simulate\_data.py}}.

\end{description}

\end{description}

\end{description}
\index{run\_script() (in module generate\_genotype\_corpus)@\spxentry{run\_script()}\spxextra{in module generate\_genotype\_corpus}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{generate_genotype_corpus:generate_genotype_corpus.run_script}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{generate\_genotype\_corpus.}}\sphinxbfcode{\sphinxupquote{run\_script}}}{}{}
Runs the script.

\end{fulllineitems}



\section{The script \sphinxstyleliteralintitle{\sphinxupquote{merge\_genotype\_corpora.py}}}
\label{\detokenize{merge_genotype_corpora:module-merge_genotype_corpora}}\label{\detokenize{merge_genotype_corpora:the-script-merge-genotype-corpora-py}}\label{\detokenize{merge_genotype_corpora::doc}}\index{merge\_genotype\_corpora (module)@\spxentry{merge\_genotype\_corpora}\spxextra{module}}
Run this script to merge pre-computed genotype corpora.

For each HAPMAP3 population and each chromosome, EpiGEN comes with a pre-computed corpus for 10000 individuals.
The corpora for chromosome \textless{}CHROM\textgreater{} have corpus ID \textless{}CHROM\textgreater{} and can be used by \sphinxcode{\sphinxupquote{simulate\_data.py}} right away.
If you want to these or other corpora, run this script.

\sphinxstylestrong{Usage}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{python3} \PYG{n}{merge\PYGZus{}genotype\PYGZus{}corpora}\PYG{o}{.}\PYG{n}{py} \PYG{p}{[}\PYG{n}{required} \PYG{n}{arguments}\PYG{p}{]} \PYG{p}{[}\PYG{n}{optional} \PYG{n}{arguments}\PYG{p}{]}
\end{sphinxVerbatim}
\begin{description}
\item[{\sphinxstylestrong{Requried Arguments:}}] \leavevmode\begin{description}
\item[{\sphinxcode{\sphinxupquote{-{-}corpus-id CORPUS\_ID}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
ID of merged genotype corpus.

\item[{Accepted Arguments:}] \leavevmode
Non-negative integers. If contained in range(1,23), the pre-computed corpora shipped with EpiGEN are overwritten.

\item[{Effect:}] \leavevmode
Together with \sphinxcode{\sphinxupquote{-{-}pop}}, this option determines the prefix \sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}}} of the files
that contain the generated corpus.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-{-}pops POP POP {[}POP ...{]}}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
List of HAPMAP3 population codes of the corpora that should be merged.
The size must match the size of the argument passed to \sphinxcode{\sphinxupquote{-{-}corpus-ids}}.

\item[{Accepted Arguments:}] \leavevmode
ASW, CEU, CEU+TSI, CHD, GIH, JPT+CHB, LWK, MEX, MKK, TSI, and MIX.

\item[{Effect:}] \leavevmode
Selects the corpora that should be merged and determines the prefix \sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}}} of the files
that contain the generated corpus. If the list passed to this argument contains only one population code,
\sphinxcode{\sphinxupquote{\textless{}POP\textgreater{}}} is the set to this code. Otherwise, \sphinxcode{\sphinxupquote{\textless{}POP\textgreater{}}} is set to \sphinxcode{\sphinxupquote{MIX}}.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-{-}corpus-ids CORPUS\_ID CORPUS\_ID {[}CORPUS\_ID ...{]}}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
The IDs of the corpora that should be merged.

\item[{Accepted Arguments:}] \leavevmode
Lists of non-negative integers of size at least 2.
The size must match the size of the argument passed to \sphinxcode{\sphinxupquote{-{-}pops}}.

\item[{Effect:}] \leavevmode
Selects the corpora that should be merged.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-{-}append APPEND}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
The axis along which the corpora should be merged.

\item[{Accepted Arguments:}] \leavevmode
“SNPS” or “INDS”.

\item[{Effect:}] \leavevmode
Set to “SNPS” to append the SNPs and to “INDS” to append the individuals.

\end{description}

\end{description}

\item[{\sphinxstylestrong{Optional Arguments:}}] \leavevmode\begin{description}
\item[{\sphinxcode{\sphinxupquote{-{-}compress}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
Compress the generated output files.

\item[{Accepted Arguments:}] \leavevmode
None.

\item[{Effect:}] \leavevmode
Determines the suffix \sphinxcode{\sphinxupquote{\textless{}SUFFIX\textgreater{}}} of the generated files. If provided, \sphinxcode{\sphinxupquote{\textless{}SUFFIX\textgreater{}}} is set to \sphinxcode{\sphinxupquote{json.bz2}}.
Otherwise, it is set to \sphinxcode{\sphinxupquote{json}}.

\end{description}

\item[{\sphinxcode{\sphinxupquote{-h, -{-}help}}}] \leavevmode\begin{description}
\item[{Effect:}] \leavevmode
Show help message and exit.

\end{description}

\end{description}

\item[{\sphinxstylestrong{Output:}}] \leavevmode\begin{description}
\item[{\sphinxstyleemphasis{Genotype Data:}}] \leavevmode\begin{description}
\item[{File:}] \leavevmode
\sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}\_genotype.\textless{}SUFFIX\textgreater{}}}

\item[{Content and Format:}] \leavevmode
(Compressed) JSON file of the form \sphinxcode{\sphinxupquote{{[}{[}G\_0\_0, ..., G\_0\_\textless{}INDS-1\textgreater{}{]}, ..., {[}G\_\textless{}SNPS-1\textgreater{}\_0, ..., G\_\textless{}SNPS-1\textgreater{}\_\textless{}INDS-1\textgreater{}{]}{]}}},
where \sphinxcode{\sphinxupquote{G\_S\_I}} encodes the number of minor alleles of the individual with index \sphinxcode{\sphinxupquote{I}} at the SNP with index \sphinxcode{\sphinxupquote{S}}.

\end{description}

\item[{\sphinxstyleemphasis{SNPs:}}] \leavevmode\begin{description}
\item[{File:}] \leavevmode
\sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}\_snps.\textless{}SUFFIX\textgreater{}}}

\item[{Content and Format:}] \leavevmode
(Compressed) JSON file of the form \sphinxcode{\sphinxupquote{{[}INFO\_0, ..., INFO\_\textless{}SNPS-1\textgreater{}{]}}}, where \sphinxcode{\sphinxupquote{INFO\_S}} contains the following
information about the SNP with index \sphinxcode{\sphinxupquote{S}}: RS identifier, chromosome number, position on chromosome, major allele, minor allele.

\end{description}

\item[{\sphinxstyleemphasis{MAFs:}}] \leavevmode\begin{description}
\item[{File:}] \leavevmode
\sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}\_mafs.\textless{}SUFFIX\textgreater{}}}

\item[{Content and Format:}] \leavevmode
(Compressed) JSON file of the form \sphinxcode{\sphinxupquote{{[}MAF\_0, ..., MAF\_\textless{}SNPS-1\textgreater{}{]}}}, where \sphinxcode{\sphinxupquote{MAF\_S}} encodes the MAF of
the SNP with index \sphinxcode{\sphinxupquote{S}}.

\end{description}

\item[{\sphinxstyleemphasis{Cumulative MAF distribution:}}] \leavevmode\begin{description}
\item[{File:}] \leavevmode
\sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}\_cum\_mafs.\textless{}SUFFIX\textgreater{}}}

\item[{Content and Format:}] \leavevmode
(Compressed) ordered JSON file of the form \sphinxcode{\sphinxupquote{{[}{[}MAF\_0, COUNT\_0{]}, ..., {[}MAF\_\textless{}NMAFS-1\textgreater{}, COUNT\_\textless{}NMAFS-1\textgreater{}{]}{]}}}, where \sphinxcode{\sphinxupquote{COUNT\_POS}} encodes the
number of SNPs is does not exceed \sphinxcode{\sphinxupquote{MAF\_POS}}.

\end{description}

\item[{\sphinxstyleemphasis{Plot of Cumulative MAF distribution:}}] \leavevmode\begin{description}
\item[{File:}] \leavevmode
\sphinxcode{\sphinxupquote{./corpora/\textless{}CORPUS\_ID\textgreater{}\_\textless{}POP\textgreater{}\_cum\_mafs.pdf}}

\item[{Content and Format:}] \leavevmode
Plot of cumulative MAF distribution. Can be used to determine feasible ranges of MAFs passed to the options
\sphinxcode{\sphinxupquote{-{-}global-maf-range}} and \sphinxcode{\sphinxupquote{-{-}disease-maf-range}} of the script \sphinxcode{\sphinxupquote{simulate\_data.py}}.

\end{description}

\end{description}

\end{description}
\index{run\_script() (in module merge\_genotype\_corpora)@\spxentry{run\_script()}\spxextra{in module merge\_genotype\_corpora}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{merge_genotype_corpora:merge_genotype_corpora.run_script}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{merge\_genotype\_corpora.}}\sphinxbfcode{\sphinxupquote{run\_script}}}{}{}
Runs the script.

\end{fulllineitems}



\section{The script \sphinxstyleliteralintitle{\sphinxupquote{test\_runtime.py}}}
\label{\detokenize{test_runtime:module-test_runtime}}\label{\detokenize{test_runtime:the-script-test-runtime-py}}\label{\detokenize{test_runtime::doc}}\index{test\_runtime (module)@\spxentry{test\_runtime}\spxextra{module}}
Run this script to test to carry out runtime tests.

\sphinxstylestrong{Usage}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{python3} \PYG{n}{test\PYGZus{}runtime}\PYG{o}{.}\PYG{n}{py}
\end{sphinxVerbatim}
\begin{description}
\item[{\sphinxstylestrong{Output:}}] \leavevmode\begin{description}
\item[{File:}] \leavevmode
\sphinxcode{\sphinxupquote{./num\_inds\_vs\_runtime.csv}}

\item[{Content:}] \leavevmode
Runtimes for simulating 1 dataset with varying number of individuals with number of SNPs fixed to 100000.

\item[{File:}] \leavevmode
\sphinxcode{\sphinxupquote{./num\_snps\_vs\_runtime.csv}}

\item[{Content:}] \leavevmode
Runtimes for simulating 1 dataset with varying number of SNPs with number of individuals fixed to 10000.

\item[{File:}] \leavevmode
\sphinxcode{\sphinxupquote{./num\_datasets\_vs\_runtime.csv}}

\item[{Content:}] \leavevmode
Runtimes for simulating varying number of datasets with number of SNPs and individuals fixed to 100.

\end{description}

\end{description}
\index{run\_script() (in module test\_runtime)@\spxentry{run\_script()}\spxextra{in module test\_runtime}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{test_runtime:test_runtime.run_script}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{test\_runtime.}}\sphinxbfcode{\sphinxupquote{run\_script}}}{}{}
Runs the script.

\end{fulllineitems}



\section{The script \sphinxstyleliteralintitle{\sphinxupquote{validate\_simulated\_data.py}}}
\label{\detokenize{validate_simulated_data:module-validate_simulated_data}}\label{\detokenize{validate_simulated_data:the-script-validate-simulated-data-py}}\label{\detokenize{validate_simulated_data::doc}}\index{validate\_simulated\_data (module)@\spxentry{validate\_simulated\_data}\spxextra{module}}
Run this script to validate the epistasis data generated with EpiGEN.

This script has to be run on top of epistasis data generated with EpiGEN.

\sphinxstylestrong{Usage}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{python3} \PYG{n}{validate\PYGZus{}simulated\PYGZus{}data}\PYG{o}{.}\PYG{n}{py} \PYG{p}{[}\PYG{n}{required} \PYG{n}{arguments}\PYG{p}{]} \PYG{p}{[}\PYG{n}{optional} \PYG{n}{arguments}\PYG{p}{]}
\end{sphinxVerbatim}
\begin{description}
\item[{\sphinxstylestrong{Required Arguments:}}] \leavevmode\begin{description}
\item[{\sphinxcode{\sphinxupquote{-{-}sim-data SIM\_DATA}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
Path to (compressed) JSON file generated by EpiGEN.

\item[{Accepted Arguments:}] \leavevmode
Strings ending in \sphinxcode{\sphinxupquote{.json}} or \sphinxcode{\sphinxupquote{.json.bz2}} that represent paths to existing result files.

\item[{Effect:}] \leavevmode
Specifies the epistasis data that should be validated.

\end{description}

\end{description}

\item[{\sphinxstylestrong{Optional Arguments:}}] \leavevmode\begin{description}
\item[{\sphinxcode{\sphinxupquote{-{-}log LOG\_FILE}}}] \leavevmode\begin{description}
\item[{Description:}] \leavevmode
If provided, the results of the validation is saved as a JSON file.

\item[{Accepted Arguments:}] \leavevmode
Strings to writable paths ending in \sphinxcode{\sphinxupquote{.json}}.

\item[{Effect:}] \leavevmode
Specifies if and if so where the results of the validation should be saved.

\end{description}

\end{description}

\item[{\sphinxstylestrong{Output:}}] \leavevmode\begin{description}
\item[{\sphinxstyleemphasis{Output File:}}] \leavevmode\begin{description}
\item[{Path:}] \leavevmode
\sphinxcode{\sphinxupquote{\textless{}LOG\_FILE\textgreater{}}}

\item[{Format:}] \leavevmode
JSON field of the form \sphinxcode{\sphinxupquote{\{"test": \textless{}TEST\textgreater{}, "p\_value": \textless{}P\_VALUE\textgreater{}, "disease\_mafs": \textless{}DISEASE\_MAFS\textgreater{}, "penetrance\_table": \textless{}PENETRANCE\_TABLE\textgreater{}\}}}.

\end{description}

\item[{\sphinxcode{\sphinxupquote{\textless{}TEST\textgreater{}}}}] \leavevmode\begin{description}
\item[{Key:}] \leavevmode
\sphinxcode{\sphinxupquote{"test"}}

\item[{Content and Format:}] \leavevmode
Statistical test used by validation: Chi-square for categorical phenotypes, one-way ANOVA F-test for
quantitative phenotypes.

\end{description}

\item[{\sphinxcode{\sphinxupquote{\textless{}P\_VALUE\textgreater{}}}}] \leavevmode\begin{description}
\item[{Key:}] \leavevmode
\sphinxcode{\sphinxupquote{"p\_value"}}

\item[{Content and Format:}] \leavevmode
The obtained p-value.

\end{description}

\item[{\sphinxcode{\sphinxupquote{\textless{}DISEASE\_MAFS\textgreater{}}}}] \leavevmode\begin{description}
\item[{Key:}] \leavevmode
\sphinxcode{\sphinxupquote{"disease\_mafs"}}

\item[{Content and Format:}] \leavevmode
JSON field of the form \sphinxcode{\sphinxupquote{{[}MAF\_0, ..., MAF\_\textless{}SNPS-1\textgreater{}{]}}}, where \sphinxcode{\sphinxupquote{MAF\_S}} encodes the MAF of
the disease SNP with index \sphinxcode{\sphinxupquote{S}} in the \sphinxcode{\sphinxupquote{"disease\_snps"}} field of the input data.

\end{description}

\item[{\sphinxcode{\sphinxupquote{\textless{}PENETRANCE\_TABLE\textgreater{}}}}] \leavevmode\begin{description}
\item[{Key:}] \leavevmode
\sphinxcode{\sphinxupquote{"penetrance\_table"}}

\item[{Content and Format:}] \leavevmode
The penetrance table underlying the test in the form \sphinxcode{\sphinxupquote{\{\textless{}GENOTYPE\textgreater{}: \textless{}PHENOTYPES\textgreater{}\}}}, where \sphinxcode{\sphinxupquote{GENOTYPE}} is
the genotype at the disease SNPs and \sphinxtitleref{PHENOTYPES{}`} is the vector of phenotypes of all individuals with the
given phenotype at the disease SNPs.{}`

\end{description}

\end{description}

\end{description}
\index{run\_script() (in module validate\_simulated\_data)@\spxentry{run\_script()}\spxextra{in module validate\_simulated\_data}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{validate_simulated_data:validate_simulated_data.run_script}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{validate\_simulated\_data.}}\sphinxbfcode{\sphinxupquote{run\_script}}}{}{}
Runs the script.

\end{fulllineitems}



\section{The package \sphinxstyleliteralintitle{\sphinxupquote{utils}}}
\label{\detokenize{utils:the-package-utils}}\label{\detokenize{utils::doc}}
This package contains the core of EpiGEN. Do not modify it, unless you really know what you are doing.


\subsection{The module \sphinxstyleliteralintitle{\sphinxupquote{utils.data\_simulator.py}}}
\label{\detokenize{utils:module-utils.data_simulator}}\label{\detokenize{utils:the-module-utils-data-simulator-py}}\index{utils.data\_simulator (module)@\spxentry{utils.data\_simulator}\spxextra{module}}
Contains definition of DataSimulator class.
\index{DataSimulator (class in utils.data\_simulator)@\spxentry{DataSimulator}\spxextra{class in utils.data\_simulator}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{utils.data\_simulator.}}\sphinxbfcode{\sphinxupquote{DataSimulator}}}{\emph{corpus\_id}, \emph{pop}, \emph{model}, \emph{num\_snps}, \emph{num\_inds}, \emph{disease\_snps}, \emph{biased\_distr}, \emph{noise\_maf\_range}, \emph{disease\_maf\_range}, \emph{seed}, \emph{compress}}{}
Bases: \sphinxcode{\sphinxupquote{object}}

Simulates epistasis data given a pre-generated genotype corpus.

This class is employed by the script simulate\_data.py.
Not intended for use outside of this script.
Expects to be imported from EpiGEN’s root directory.
\index{genotype (utils.data\_simulator.DataSimulator attribute)@\spxentry{genotype}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.genotype}}\pysigline{\sphinxbfcode{\sphinxupquote{genotype}}}
A numpy.array with entries from range(3) whose rows represent SNPs and whose
columns represent individuals contained in simulated data.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
numpy.array

\end{description}\end{quote}

\end{fulllineitems}

\index{snps (utils.data\_simulator.DataSimulator attribute)@\spxentry{snps}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.snps}}\pysigline{\sphinxbfcode{\sphinxupquote{snps}}}
A list with one entry for each row of self.genotype. The entries provide information
about the corresponding SNP.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
list of (list of str)

\end{description}\end{quote}

\end{fulllineitems}

\index{mafs (utils.data\_simulator.DataSimulator attribute)@\spxentry{mafs}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.mafs}}\pysigline{\sphinxbfcode{\sphinxupquote{mafs}}}
A numpy.array of floats representing the MAFs of all rows of self.genotype.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
numpy.array

\end{description}\end{quote}

\end{fulllineitems}

\index{cum\_mafs (utils.data\_simulator.DataSimulator attribute)@\spxentry{cum\_mafs}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.cum_mafs}}\pysigline{\sphinxbfcode{\sphinxupquote{cum\_mafs}}}
A list of pairs of the form (MAF,count) representing the cumulative MAF distribution of self.mafs.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
list of (float,int)

\end{description}\end{quote}

\end{fulllineitems}

\index{corpus\_genotype (utils.data\_simulator.DataSimulator attribute)@\spxentry{corpus\_genotype}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.corpus_genotype}}\pysigline{\sphinxbfcode{\sphinxupquote{corpus\_genotype}}}
A numpy.array with entries from range(3) whose rows represent SNPs and whose
columns represent individuals contained in the genotype corpus.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
numpy.array

\end{description}\end{quote}

\end{fulllineitems}

\index{corpus\_snps (utils.data\_simulator.DataSimulator attribute)@\spxentry{corpus\_snps}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.corpus_snps}}\pysigline{\sphinxbfcode{\sphinxupquote{corpus\_snps}}}
A list with one entry for each row of self.corpus\_genotype. The entries provide information
about the corresponding SNP.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
list of (list of str)

\end{description}\end{quote}

\end{fulllineitems}

\index{corpus\_mafs (utils.data\_simulator.DataSimulator attribute)@\spxentry{corpus\_mafs}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.corpus_mafs}}\pysigline{\sphinxbfcode{\sphinxupquote{corpus\_mafs}}}
A numpy.array of floats representing the MAFs of all rows of self.corpus\_genotype.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
numpy.array

\end{description}\end{quote}

\end{fulllineitems}

\index{corpus\_cum\_mafs (utils.data\_simulator.DataSimulator attribute)@\spxentry{corpus\_cum\_mafs}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.corpus_cum_mafs}}\pysigline{\sphinxbfcode{\sphinxupquote{corpus\_cum\_mafs}}}
A list of pairs of the form (MAF,count) representing the cumulative MAF distribution of self.corpus\_mafs.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
list of (float,int)

\end{description}\end{quote}

\end{fulllineitems}

\index{model (utils.data\_simulator.DataSimulator attribute)@\spxentry{model}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.model}}\pysigline{\sphinxbfcode{\sphinxupquote{model}}}
The epistasis model.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
ExtensionalModel/ParametrizedModel

\end{description}\end{quote}

\end{fulllineitems}

\index{sim\_id (utils.data\_simulator.DataSimulator attribute)@\spxentry{sim\_id}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.sim_id}}\pysigline{\sphinxbfcode{\sphinxupquote{sim\_id}}}
An integer that represents the ID of the generated data.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int

\end{description}\end{quote}

\end{fulllineitems}

\index{corpus\_id (utils.data\_simulator.DataSimulator attribute)@\spxentry{corpus\_id}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.corpus_id}}\pysigline{\sphinxbfcode{\sphinxupquote{corpus\_id}}}
An integer that represents the ID of the genotype corpus on top of which
the data should be simulated.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int

\end{description}\end{quote}

\end{fulllineitems}

\index{pop (utils.data\_simulator.DataSimulator attribute)@\spxentry{pop}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.pop}}\pysigline{\sphinxbfcode{\sphinxupquote{pop}}}
A string representing the HAPMAP3 population for which the selected genotype corpus was
generated.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
str

\end{description}\end{quote}

\end{fulllineitems}

\index{num\_snps (utils.data\_simulator.DataSimulator attribute)@\spxentry{num\_snps}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.num_snps}}\pysigline{\sphinxbfcode{\sphinxupquote{num\_snps}}}
The number of SNPs in the simulated data.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int

\end{description}\end{quote}

\end{fulllineitems}

\index{num\_inds (utils.data\_simulator.DataSimulator attribute)@\spxentry{num\_inds}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.num_inds}}\pysigline{\sphinxbfcode{\sphinxupquote{num\_inds}}}
The number of individuals in the simulated data.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int

\end{description}\end{quote}

\end{fulllineitems}

\index{total\_num\_snps (utils.data\_simulator.DataSimulator attribute)@\spxentry{total\_num\_snps}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.total_num_snps}}\pysigline{\sphinxbfcode{\sphinxupquote{total\_num\_snps}}}
The number of SNPs in the selected genotype corpus.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int

\end{description}\end{quote}

\end{fulllineitems}

\index{total\_num\_inds (utils.data\_simulator.DataSimulator attribute)@\spxentry{total\_num\_inds}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.total_num_inds}}\pysigline{\sphinxbfcode{\sphinxupquote{total\_num\_inds}}}
The number of individuals in the selected genotype corpus.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int

\end{description}\end{quote}

\end{fulllineitems}

\index{biased\_distr (utils.data\_simulator.DataSimulator attribute)@\spxentry{biased\_distr}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.biased_distr}}\pysigline{\sphinxbfcode{\sphinxupquote{biased\_distr}}}
Parameters of biased observed phenotype distribution. If empty, no observation bias is applied.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
list of float

\end{description}\end{quote}

\end{fulllineitems}

\index{phenotype (utils.data\_simulator.DataSimulator attribute)@\spxentry{phenotype}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.phenotype}}\pysigline{\sphinxbfcode{\sphinxupquote{phenotype}}}
A numpy.array that stores the generated phenotypes.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
numpy.array

\end{description}\end{quote}

\end{fulllineitems}

\index{disease\_snps (utils.data\_simulator.DataSimulator attribute)@\spxentry{disease\_snps}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.disease_snps}}\pysigline{\sphinxbfcode{\sphinxupquote{disease\_snps}}}
A list of positions of the selected disease SNPs in self.snps and self.genotype.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
list of int

\end{description}\end{quote}

\end{fulllineitems}

\index{input\_disease\_snps (utils.data\_simulator.DataSimulator attribute)@\spxentry{input\_disease\_snps}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.input_disease_snps}}\pysigline{\sphinxbfcode{\sphinxupquote{input\_disease\_snps}}}
A user-specified list of positions of the selected disease SNPs in self.snps and self.genotype.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
list of int

\end{description}\end{quote}

\end{fulllineitems}

\index{noise\_maf\_range (utils.data\_simulator.DataSimulator attribute)@\spxentry{noise\_maf\_range}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.noise_maf_range}}\pysigline{\sphinxbfcode{\sphinxupquote{noise\_maf\_range}}}
A tuple of floats between 0 and 1 that specifies the range of acceptable MAFs for the selected noise SNPs.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
float,float

\end{description}\end{quote}

\end{fulllineitems}

\index{disease\_maf\_range (utils.data\_simulator.DataSimulator attribute)@\spxentry{disease\_maf\_range}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.disease_maf_range}}\pysigline{\sphinxbfcode{\sphinxupquote{disease\_maf\_range}}}
A tuple of floats between 0 and 1 that specifies the range of acceptable MAFs for the selected disease SNPs.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
float,float

\end{description}\end{quote}

\end{fulllineitems}

\index{epsilon (utils.data\_simulator.DataSimulator attribute)@\spxentry{epsilon}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.epsilon}}\pysigline{\sphinxbfcode{\sphinxupquote{epsilon}}}
A small positive real used for comparing floats.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
float

\end{description}\end{quote}

\end{fulllineitems}

\index{compress (utils.data\_simulator.DataSimulator attribute)@\spxentry{compress}\spxextra{utils.data\_simulator.DataSimulator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.compress}}\pysigline{\sphinxbfcode{\sphinxupquote{compress}}}
If True, the simulated data is compressed.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
bool

\end{description}\end{quote}

\end{fulllineitems}

\index{\_\_init\_\_() (utils.data\_simulator.DataSimulator method)@\spxentry{\_\_init\_\_()}\spxextra{utils.data\_simulator.DataSimulator method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.__init__}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{\_\_init\_\_}}}{\emph{corpus\_id}, \emph{pop}, \emph{model}, \emph{num\_snps}, \emph{num\_inds}, \emph{disease\_snps}, \emph{biased\_distr}, \emph{noise\_maf\_range}, \emph{disease\_maf\_range}, \emph{seed}, \emph{compress}}{}
Initialized DataSimulator.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{corpus\_id}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} An integer that represents the ID of the genotype corpus on top of which
the data should be simulated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{pop}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} A string representing the HAPMAP3 population for which the selected genotype corpus was
generated.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{model}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Path to an INI or an XML file containing the epistasis model specification.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{num\_snps}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} The number of SNPs in the simulated data.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{num\_inds}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} The number of individuals in the simulated data.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{disease\_snps}} (\sphinxstyleliteralemphasis{\sphinxupquote{list of int}}) \textendash{} A list the disease SNPs’ position in the selected genotype corpus. If not empty, its size must match the size of the model.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{biased\_distr}} (\sphinxstyleliteralemphasis{\sphinxupquote{list of float}}) \textendash{} Parameters of biased observed phenotype distribution. If empty, no observation bias is applied.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{noise\_maf\_range}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{,}}\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} A tuple of floats between 0 and 1 that specifies the range of acceptable MAFs for the selected noise SNPs.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{disease\_maf\_range}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{,}}\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} A tuple of floats between 0 and 1 that specifies the range of acceptable MAFs for the selected disease SNPs.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{seed}} (\sphinxstyleliteralemphasis{\sphinxupquote{int/None}}) \textendash{} The seed for numpy.random (possibly None).

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{compress}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} If True, the simulated data is compressed.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{dump\_simulated\_data() (utils.data\_simulator.DataSimulator method)@\spxentry{dump\_simulated\_data()}\spxextra{utils.data\_simulator.DataSimulator method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.dump_simulated_data}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dump\_simulated\_data}}}{}{}
Dumps the simulated data.

\end{fulllineitems}

\index{generate\_phenotype() (utils.data\_simulator.DataSimulator method)@\spxentry{generate\_phenotype()}\spxextra{utils.data\_simulator.DataSimulator method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.generate_phenotype}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_phenotype}}}{}{}
Generates the phenotype and adjusts the number of individuals.

\end{fulllineitems}

\index{sample\_snps() (utils.data\_simulator.DataSimulator method)@\spxentry{sample\_snps()}\spxextra{utils.data\_simulator.DataSimulator method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.sample_snps}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{sample\_snps}}}{}{}
Samples the SNPs and the disease SNP set based on the MAFs.

\end{fulllineitems}

\index{set\_sim\_id() (utils.data\_simulator.DataSimulator method)@\spxentry{set\_sim\_id()}\spxextra{utils.data\_simulator.DataSimulator method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.data_simulator.DataSimulator.set_sim_id}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_sim\_id}}}{\emph{sim\_id}}{}
Sets the simulation ID and prepares next simulation on top of pre-loaded corpus.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{sim\_id}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} An integer that represents the ID of the generated data.

\end{description}\end{quote}

\end{fulllineitems}


\end{fulllineitems}



\subsection{The module \sphinxstyleliteralintitle{\sphinxupquote{utils.genotype\_corpus\_generator.py}}}
\label{\detokenize{utils:module-utils.genotype_corpus_generator}}\label{\detokenize{utils:the-module-utils-genotype-corpus-generator-py}}\index{utils.genotype\_corpus\_generator (module)@\spxentry{utils.genotype\_corpus\_generator}\spxextra{module}}
Contains definition of GenotypeCorpusGenerator class.
\index{GenotypeCorpusGenerator (class in utils.genotype\_corpus\_generator)@\spxentry{GenotypeCorpusGenerator}\spxextra{class in utils.genotype\_corpus\_generator}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{utils.genotype\_corpus\_generator.}}\sphinxbfcode{\sphinxupquote{GenotypeCorpusGenerator}}}{\emph{chroms}, \emph{num\_inds}, \emph{corpus\_id}, \emph{pop}, \emph{compress}}{}
Bases: \sphinxcode{\sphinxupquote{object}}

Generates genotype corpus by calling HAPGEN2 and merging the obtained chromosome-wise genotypes.

This class is employed by the script generate\_genotype\_corpus.py.
Not intended for use outside of this script.
Expects to be imported from EpiGEN’s root directory.
\index{hapmap\_dir (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute)@\spxentry{hapmap\_dir}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.hapmap_dir}}\pysigline{\sphinxbfcode{\sphinxupquote{hapmap\_dir}}}
The path to the HAPMAP3 directory.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
str

\end{description}\end{quote}

\end{fulllineitems}

\index{hapgen\_dir (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute)@\spxentry{hapgen\_dir}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.hapgen_dir}}\pysigline{\sphinxbfcode{\sphinxupquote{hapgen\_dir}}}
The path to the directory that contains the HAPGEN2 binary.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
str

\end{description}\end{quote}

\end{fulllineitems}

\index{chroms (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute)@\spxentry{chroms}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.chroms}}\pysigline{\sphinxbfcode{\sphinxupquote{chroms}}}
A list of integers between 1 and 22 representing the chromosomes
for which HAPGEN2 generates the genotypes.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
list of int

\end{description}\end{quote}

\end{fulllineitems}

\index{num\_inds (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute)@\spxentry{num\_inds}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.num_inds}}\pysigline{\sphinxbfcode{\sphinxupquote{num\_inds}}}
An integer that represents the number of individuals for which
genotypes are constructed.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int

\end{description}\end{quote}

\end{fulllineitems}

\index{num\_snps (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute)@\spxentry{num\_snps}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.num_snps}}\pysigline{\sphinxbfcode{\sphinxupquote{num\_snps}}}
An integer that represents the number of SNPs in the generated corpus.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int

\end{description}\end{quote}

\end{fulllineitems}

\index{corpus\_id (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute)@\spxentry{corpus\_id}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.corpus_id}}\pysigline{\sphinxbfcode{\sphinxupquote{corpus\_id}}}
An integer that represents the ID of the generated corpus.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int

\end{description}\end{quote}

\end{fulllineitems}

\index{pop (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute)@\spxentry{pop}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.pop}}\pysigline{\sphinxbfcode{\sphinxupquote{pop}}}
A string representing the HAPMAP3 population.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
str

\end{description}\end{quote}

\end{fulllineitems}

\index{dummy\_disease\_snps (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute)@\spxentry{dummy\_disease\_snps}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.dummy_disease_snps}}\pysigline{\sphinxbfcode{\sphinxupquote{dummy\_disease\_snps}}}
A dict that, for each chromosome, specifies the position
of a SNP that can be passed to HAPGEN2 as argument of the -dl option.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
dict of (int,int)

\end{description}\end{quote}

\end{fulllineitems}

\index{genotype (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute)@\spxentry{genotype}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.genotype}}\pysigline{\sphinxbfcode{\sphinxupquote{genotype}}}
A numpy.array with entries from range(3) that contains the merged genotypes.
The rows represent SNPs, the columns represent individuals.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
numpy.array

\end{description}\end{quote}

\end{fulllineitems}

\index{snps (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute)@\spxentry{snps}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.snps}}\pysigline{\sphinxbfcode{\sphinxupquote{snps}}}
A list with one entry for each row of self.genotype. The entries provide information
about the corresponding SNP.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
list of (list of str)

\end{description}\end{quote}

\end{fulllineitems}

\index{mafs (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute)@\spxentry{mafs}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.mafs}}\pysigline{\sphinxbfcode{\sphinxupquote{mafs}}}
A numpy.array of floats representing the MAFs of all rows of self.genotype.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
numpy.array

\end{description}\end{quote}

\end{fulllineitems}

\index{cum\_mafs (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute)@\spxentry{cum\_mafs}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.cum_mafs}}\pysigline{\sphinxbfcode{\sphinxupquote{cum\_mafs}}}
A list of pairs of the form (MAF,count) representing the cumulative MAF distribution.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
list of (float,int)

\end{description}\end{quote}

\end{fulllineitems}

\index{compress (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute)@\spxentry{compress}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.compress}}\pysigline{\sphinxbfcode{\sphinxupquote{compress}}}
If True, the generated corpus is compressed.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
bool

\end{description}\end{quote}

\end{fulllineitems}

\index{\_\_init\_\_() (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator method)@\spxentry{\_\_init\_\_()}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.__init__}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{\_\_init\_\_}}}{\emph{chroms}, \emph{num\_inds}, \emph{corpus\_id}, \emph{pop}, \emph{compress}}{}
Initializes GenotypeCorpusGenerator.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{chroms}} (\sphinxstyleliteralemphasis{\sphinxupquote{list of int}}) \textendash{} A list of integers between 1 and 22 representing the chroms for which
HAPGEN2 should be called. Duplicates are ignored.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{num\_inds}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} An integer representing the number of individuals for which genotypes should
be constructed.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{corpus\_id}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} An integer that represents the ID of the generated corpus.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{compress}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} If True, the generated corpus is compressed.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{call\_hapgen2() (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator method)@\spxentry{call\_hapgen2()}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.call_hapgen2}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{call\_hapgen2}}}{}{}
Calls HAPGEN2 to generate genotypes for all chromosomes.

\end{fulllineitems}

\index{compute\_mafs() (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator method)@\spxentry{compute\_mafs()}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.compute_mafs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{compute\_mafs}}}{}{}
Computes MAFs for all SNPs contained in self.genotype.

\end{fulllineitems}

\index{dump\_corpus() (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator method)@\spxentry{dump\_corpus()}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.dump_corpus}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dump\_corpus}}}{}{}
Dumps the generated corpus to zipped JSON files.

\end{fulllineitems}

\index{merge\_hapgen2\_output() (utils.genotype\_corpus\_generator.GenotypeCorpusGenerator method)@\spxentry{merge\_hapgen2\_output()}\spxextra{utils.genotype\_corpus\_generator.GenotypeCorpusGenerator method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_generator.GenotypeCorpusGenerator.merge_hapgen2_output}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{merge\_hapgen2\_output}}}{}{}
Merges the output of HAPGEN2.

\end{fulllineitems}


\end{fulllineitems}



\subsection{The module \sphinxstyleliteralintitle{\sphinxupquote{utils.genotype\_corpus\_merger.py}}}
\label{\detokenize{utils:module-utils.genotype_corpus_merger}}\label{\detokenize{utils:the-module-utils-genotype-corpus-merger-py}}\index{utils.genotype\_corpus\_merger (module)@\spxentry{utils.genotype\_corpus\_merger}\spxextra{module}}
Contains definition of GenotypeCorpusMerger class.
\index{GenotypeCorpusMerger (class in utils.genotype\_corpus\_merger)@\spxentry{GenotypeCorpusMerger}\spxextra{class in utils.genotype\_corpus\_merger}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{utils.genotype\_corpus\_merger.}}\sphinxbfcode{\sphinxupquote{GenotypeCorpusMerger}}}{\emph{corpus\_ids}, \emph{pops}, \emph{corpus\_id}, \emph{axis}, \emph{compress}}{}
Bases: \sphinxcode{\sphinxupquote{object}}

Merges pre-computed genotype corpora.

This class is employed by the script generate\_genotype\_corpus.py.
Not intended for use outside of this script.
Expects to be imported from EpiGEN’s root directory.
\index{corpus\_ids (utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute)@\spxentry{corpus\_ids}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.corpus_ids}}\pysigline{\sphinxbfcode{\sphinxupquote{corpus\_ids}}}
A list of integers that represents the IDs of the corpora that should be merged.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
list of int

\end{description}\end{quote}

\end{fulllineitems}

\index{pops (utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute)@\spxentry{pops}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.pops}}\pysigline{\sphinxbfcode{\sphinxupquote{pops}}}
A list of strings representing the HAPMAP3 population codes of the corpora that should be merged.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
list of str

\end{description}\end{quote}

\end{fulllineitems}

\index{corpus\_id (utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute)@\spxentry{corpus\_id}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.corpus_id}}\pysigline{\sphinxbfcode{\sphinxupquote{corpus\_id}}}
An integer that represents the ID of the generated corpus.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int

\end{description}\end{quote}

\end{fulllineitems}

\index{pop (utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute)@\spxentry{pop}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.pop}}\pysigline{\sphinxbfcode{\sphinxupquote{pop}}}
A string representing the HAPMAP3 population code of the merged corpus.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
str

\end{description}\end{quote}

\end{fulllineitems}

\index{genotype (utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute)@\spxentry{genotype}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.genotype}}\pysigline{\sphinxbfcode{\sphinxupquote{genotype}}}
A numpy.array with entries from range(3) that contains the merged genotypes.
The rows represent SNPs, the columns represent individuals.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
numpy.array

\end{description}\end{quote}

\end{fulllineitems}

\index{snps (utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute)@\spxentry{snps}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.snps}}\pysigline{\sphinxbfcode{\sphinxupquote{snps}}}
A list with one entry for each row of self.genotype. The entries provide information
about the corresponding SNP.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
list of (list of str)

\end{description}\end{quote}

\end{fulllineitems}

\index{mafs (utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute)@\spxentry{mafs}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.mafs}}\pysigline{\sphinxbfcode{\sphinxupquote{mafs}}}
A numpy.array of floats representing the MAFs of all rows of self.genotype.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
numpy.array

\end{description}\end{quote}

\end{fulllineitems}

\index{cum\_mafs (utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute)@\spxentry{cum\_mafs}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.cum_mafs}}\pysigline{\sphinxbfcode{\sphinxupquote{cum\_mafs}}}
A list of pairs of the form (MAF,count) representing the cumulative MAF distribution.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
list of (float,int)

\end{description}\end{quote}

\end{fulllineitems}

\index{axis (utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute)@\spxentry{axis}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.axis}}\pysigline{\sphinxbfcode{\sphinxupquote{axis}}}
An integer representing the axis of the merge (0 for merge along SNPs, 1 for merge along individuals).
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int

\end{description}\end{quote}

\end{fulllineitems}

\index{num\_snps (utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute)@\spxentry{num\_snps}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.num_snps}}\pysigline{\sphinxbfcode{\sphinxupquote{num\_snps}}}
Number of SNPs in merged corpus.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int

\end{description}\end{quote}

\end{fulllineitems}

\index{num\_inds (utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute)@\spxentry{num\_inds}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.num_inds}}\pysigline{\sphinxbfcode{\sphinxupquote{num\_inds}}}
Number of individuals in merged corpus.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int

\end{description}\end{quote}

\end{fulllineitems}

\index{compress (utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute)@\spxentry{compress}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.compress}}\pysigline{\sphinxbfcode{\sphinxupquote{compress}}}
If True, the merged corpus is compressed.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
bool

\end{description}\end{quote}

\end{fulllineitems}

\index{\_\_init\_\_() (utils.genotype\_corpus\_merger.GenotypeCorpusMerger method)@\spxentry{\_\_init\_\_()}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.__init__}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{\_\_init\_\_}}}{\emph{corpus\_ids}, \emph{pops}, \emph{corpus\_id}, \emph{axis}, \emph{compress}}{}
Initializes GenotypeCorpusGenerator.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{corpus\_ids}} (\sphinxstyleliteralemphasis{\sphinxupquote{list of int}}) \textendash{} A list of integers that represents the IDs of the corpora that should be merged.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{pop}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} A list of strings representing the HAPMAP3 population codes of the corpora that should be merged.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{corpus\_id}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} An integer that represents the ID of the generated corpus.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{axis}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} An integer representing the axis of the merge (0 for merge along SNPs, 1 for merge along individuals)

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{compress}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} If True, the merged corpus is compressed.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{compute\_mafs() (utils.genotype\_corpus\_merger.GenotypeCorpusMerger method)@\spxentry{compute\_mafs()}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.compute_mafs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{compute\_mafs}}}{}{}
Computes MAFs for all SNPs contained in self.genotype.

\end{fulllineitems}

\index{dump\_corpus() (utils.genotype\_corpus\_merger.GenotypeCorpusMerger method)@\spxentry{dump\_corpus()}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.dump_corpus}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dump\_corpus}}}{}{}
Dumps the generated corpus to zipped JSON files.

\end{fulllineitems}

\index{merge\_corpora() (utils.genotype\_corpus\_merger.GenotypeCorpusMerger method)@\spxentry{merge\_corpora()}\spxextra{utils.genotype\_corpus\_merger.GenotypeCorpusMerger method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.genotype_corpus_merger.GenotypeCorpusMerger.merge_corpora}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{merge\_corpora}}}{}{}
\end{fulllineitems}


\end{fulllineitems}



\subsection{The module \sphinxstyleliteralintitle{\sphinxupquote{utils.extensional\_model.py}}}
\label{\detokenize{utils:module-utils.extensional_model}}\label{\detokenize{utils:the-module-utils-extensional-model-py}}\index{utils.extensional\_model (module)@\spxentry{utils.extensional\_model}\spxextra{module}}
Contains definition of ExtensionalModel class.
\index{ExtensionalModel (class in utils.extensional\_model)@\spxentry{ExtensionalModel}\spxextra{class in utils.extensional\_model}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.extensional_model.ExtensionalModel}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{utils.extensional\_model.}}\sphinxbfcode{\sphinxupquote{ExtensionalModel}}}{\emph{model}, \emph{seed}}{}
Bases: \sphinxcode{\sphinxupquote{object}}

Represents an extensionally defined epistasis model.

Extensional models can be used to model binary or non-binary categorical phenotypes, as well as quantitative phenotypes.
This class is employed by the class DataSimulator.
Not intended for use outside of this class.
\index{size (utils.extensional\_model.ExtensionalModel attribute)@\spxentry{size}\spxextra{utils.extensional\_model.ExtensionalModel attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.extensional_model.ExtensionalModel.size}}\pysigline{\sphinxbfcode{\sphinxupquote{size}}}
An integer greater equal 2 representing the size of the model.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int

\end{description}\end{quote}

\end{fulllineitems}

\index{model (utils.extensional\_model.ExtensionalModel attribute)@\spxentry{model}\spxextra{utils.extensional\_model.ExtensionalModel attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.extensional_model.ExtensionalModel.model}}\pysigline{\sphinxbfcode{\sphinxupquote{model}}}
A numpy.array of dimension self.size representing the epistasis model.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
numpy.array

\end{description}\end{quote}

\end{fulllineitems}

\index{phenotype (utils.extensional\_model.ExtensionalModel attribute)@\spxentry{phenotype}\spxextra{utils.extensional\_model.ExtensionalModel attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.extensional_model.ExtensionalModel.phenotype}}\pysigline{\sphinxbfcode{\sphinxupquote{phenotype}}}
An integer greater equal 2 (for categorical phenotypes)
or the string “quantitative” (for quantitative phenotypes).
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int/str

\end{description}\end{quote}

\end{fulllineitems}

\index{\_\_init\_\_() (utils.extensional\_model.ExtensionalModel method)@\spxentry{\_\_init\_\_()}\spxextra{utils.extensional\_model.ExtensionalModel method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.extensional_model.ExtensionalModel.__init__}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{\_\_init\_\_}}}{\emph{model}, \emph{seed}}{}
Initializes the ExtensionalModel.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{model}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Path to an INI file that contains the full extensional definiton of an epistasis model.
For categorical models, a discrete probability distribution must be provided for each possible genotype of dimension self.size.
For quantitative models, the mean and the standard deviation of normal distributions must be provided.
Examples can be found in the directory epigen/models/.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{seed}} (\sphinxstyleliteralemphasis{\sphinxupquote{int/None}}) \textendash{} The seed for np.random (possibly None).

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}


\end{fulllineitems}



\subsection{The module \sphinxstyleliteralintitle{\sphinxupquote{utils.parametrized\_model.py}}}
\label{\detokenize{utils:module-utils.parametrized_model}}\label{\detokenize{utils:the-module-utils-parametrized-model-py}}\index{utils.parametrized\_model (module)@\spxentry{utils.parametrized\_model}\spxextra{module}}
Contains definition of ParametrizedModel class.
\index{ParametrizedModel (class in utils.parametrized\_model)@\spxentry{ParametrizedModel}\spxextra{class in utils.parametrized\_model}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.parametrized_model.ParametrizedModel}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{utils.parametrized\_model.}}\sphinxbfcode{\sphinxupquote{ParametrizedModel}}}{\emph{model}, \emph{seed}}{}
Bases: \sphinxcode{\sphinxupquote{object}}

Represents a parametrized epistasis model.

Parametrized models can be used to model binary categorical phenotypes.
This class is employed by the class DataSimulator.
Not intended for use outside of this class.
\index{size (utils.parametrized\_model.ParametrizedModel attribute)@\spxentry{size}\spxextra{utils.parametrized\_model.ParametrizedModel attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.parametrized_model.ParametrizedModel.size}}\pysigline{\sphinxbfcode{\sphinxupquote{size}}}
An integer greater equal 2 representing the size of the model.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int

\end{description}\end{quote}

\end{fulllineitems}

\index{baseline\_alpha (utils.parametrized\_model.ParametrizedModel attribute)@\spxentry{baseline\_alpha}\spxextra{utils.parametrized\_model.ParametrizedModel attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.parametrized_model.ParametrizedModel.baseline_alpha}}\pysigline{\sphinxbfcode{\sphinxupquote{baseline\_alpha}}}
A float greater 0 representing the baseline risk.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
float

\end{description}\end{quote}

\end{fulllineitems}

\index{marginal\_models (utils.parametrized\_model.ParametrizedModel attribute)@\spxentry{marginal\_models}\spxextra{utils.parametrized\_model.ParametrizedModel attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.parametrized_model.ParametrizedModel.marginal_models}}\pysigline{\sphinxbfcode{\sphinxupquote{marginal\_models}}}
The marginal models.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
dict of (int,callable)

\end{description}\end{quote}

\end{fulllineitems}

\index{interaction\_models (utils.parametrized\_model.ParametrizedModel attribute)@\spxentry{interaction\_models}\spxextra{utils.parametrized\_model.ParametrizedModel attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.parametrized_model.ParametrizedModel.interaction_models}}\pysigline{\sphinxbfcode{\sphinxupquote{interaction\_models}}}
The interaction models.
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
dict of ((list of int),callable)

\end{description}\end{quote}

\end{fulllineitems}

\index{phenotype (utils.parametrized\_model.ParametrizedModel attribute)@\spxentry{phenotype}\spxextra{utils.parametrized\_model.ParametrizedModel attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.parametrized_model.ParametrizedModel.phenotype}}\pysigline{\sphinxbfcode{\sphinxupquote{phenotype}}}
The inetger 2 (for dichotomous phenotypes) or the
string “quantitative” (for quantitative phenotypes).
\begin{quote}\begin{description}
\item[{Type}] \leavevmode
int/str

\end{description}\end{quote}

\end{fulllineitems}

\index{\_\_init\_\_() (utils.parametrized\_model.ParametrizedModel method)@\spxentry{\_\_init\_\_()}\spxextra{utils.parametrized\_model.ParametrizedModel method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.parametrized_model.ParametrizedModel.__init__}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{\_\_init\_\_}}}{\emph{model}, \emph{seed}}{}
Initializes the ExtensionalModel.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{model}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Path to an XML file that contains the specification of the parametrized model.
Examples can be found in the directory epigen/models/.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{seed}} (\sphinxstyleliteralemphasis{\sphinxupquote{int/None}}) \textendash{} The seed for np.random (possibly None).

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}


\end{fulllineitems}



\subsection{The module \sphinxstyleliteralintitle{\sphinxupquote{utils.argparse\_checks.py}}}
\label{\detokenize{utils:module-utils.argparse_checks}}\label{\detokenize{utils:the-module-utils-argparse-checks-py}}\index{utils.argparse\_checks (module)@\spxentry{utils.argparse\_checks}\spxextra{module}}\index{check\_interval() (in module utils.argparse\_checks)@\spxentry{check\_interval()}\spxextra{in module utils.argparse\_checks}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.argparse_checks.check_interval}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.argparse\_checks.}}\sphinxbfcode{\sphinxupquote{check\_interval}}}{\emph{argname}}{}
Ensures that the provided arguments specify a sub-interval of {[}0,1{]}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{argname}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the argparse argument.

\end{description}\end{quote}

\end{fulllineitems}

\index{check\_length() (in module utils.argparse\_checks)@\spxentry{check\_length()}\spxextra{in module utils.argparse\_checks}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.argparse_checks.check_length}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.argparse\_checks.}}\sphinxbfcode{\sphinxupquote{check\_length}}}{\emph{argname}}{}
Ensures that at least two arguments are provided.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{argname}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the argparse argument.

\end{description}\end{quote}

\end{fulllineitems}

\index{check\_non\_negative() (in module utils.argparse\_checks)@\spxentry{check\_non\_negative()}\spxextra{in module utils.argparse\_checks}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.argparse_checks.check_non_negative}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.argparse\_checks.}}\sphinxbfcode{\sphinxupquote{check\_non\_negative}}}{\emph{argname}}{}
Ensures that the provided argument is non-negative.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{argname}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the argparse argument.

\end{description}\end{quote}

\end{fulllineitems}

\index{check\_positive() (in module utils.argparse\_checks)@\spxentry{check\_positive()}\spxextra{in module utils.argparse\_checks}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.argparse_checks.check_positive}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.argparse\_checks.}}\sphinxbfcode{\sphinxupquote{check\_positive}}}{\emph{argname}}{}
Ensures that the provided argument is positive.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{argname}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the argparse argument.

\end{description}\end{quote}

\end{fulllineitems}



\subsection{The module \sphinxstyleliteralintitle{\sphinxupquote{utils.validation\_utils.py}}}
\label{\detokenize{utils:module-utils.validation_utils}}\label{\detokenize{utils:the-module-utils-validation-utils-py}}\index{utils.validation\_utils (module)@\spxentry{utils.validation\_utils}\spxextra{module}}
Contains utility functions for validating simulated data.
\index{chi\_square() (in module utils.validation\_utils)@\spxentry{chi\_square()}\spxextra{in module utils.validation\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.validation_utils.chi_square}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.validation\_utils.}}\sphinxbfcode{\sphinxupquote{chi\_square}}}{\emph{penetrance\_table}, \emph{num\_categories}}{}
Carries out chi-square test.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{penetrance\_table}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} The penetrance table induced by the disease SNPs.

\item[{Returns}] \leavevmode
The obtained p-value.

\item[{Return type}] \leavevmode
float

\end{description}\end{quote}

\end{fulllineitems}

\index{generate\_penetrance\_table() (in module utils.validation\_utils)@\spxentry{generate\_penetrance\_table()}\spxextra{in module utils.validation\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.validation_utils.generate_penetrance_table}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.validation\_utils.}}\sphinxbfcode{\sphinxupquote{generate\_penetrance\_table}}}{\emph{sim\_data}}{}
Generates penetrance table.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{sim\_data}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} The simulated data.

\item[{Returns}] \leavevmode
The penetrance table induced by the disease SNPs.

\item[{Return type}] \leavevmode
dict

\end{description}\end{quote}

\end{fulllineitems}

\index{load\_data() (in module utils.validation\_utils)@\spxentry{load\_data()}\spxextra{in module utils.validation\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.validation_utils.load_data}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.validation\_utils.}}\sphinxbfcode{\sphinxupquote{load\_data}}}{\emph{filename}}{}
Loads simulated data.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{filename}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Path to the file containing the data.

\item[{Returns}] \leavevmode
A dictionary containing the simulated data.

\item[{Return type}] \leavevmode
dict

\end{description}\end{quote}

\end{fulllineitems}

\index{one\_way\_anova() (in module utils.validation\_utils)@\spxentry{one\_way\_anova()}\spxextra{in module utils.validation\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.validation_utils.one_way_anova}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.validation\_utils.}}\sphinxbfcode{\sphinxupquote{one\_way\_anova}}}{\emph{penetrance\_table}}{}
Carries out one-way ANOVA F-test.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\sphinxstyleliteralstrong{\sphinxupquote{penetrance\_table}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} The penetrance table induced by the disease SNPs.

\item[{Returns}] \leavevmode
The obtained p-value.

\item[{Return type}] \leavevmode
float

\end{description}\end{quote}

\end{fulllineitems}

\index{write\_to\_log\_file() (in module utils.validation\_utils)@\spxentry{write\_to\_log\_file()}\spxextra{in module utils.validation\_utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.validation_utils.write_to_log_file}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.validation\_utils.}}\sphinxbfcode{\sphinxupquote{write\_to\_log\_file}}}{\emph{logfilename}, \emph{test}, \emph{p\_value}, \emph{disease\_mafs}, \emph{penetrance\_table}}{}
Writes the results of the validation to a log-file.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{logfilename}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the log-file.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{test}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the test.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{p\_value}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} Obtained p-value.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{disease\_mafs}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of MAFs of disease SNPs.

\item {} 
\sphinxstyleliteralstrong{\sphinxupquote{penetrance\_table}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Penetrance table induced by disease SNPs.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}



\renewcommand{\indexname}{Python Module Index}
\begin{sphinxtheindex}
\let\bigletter\sphinxstyleindexlettergroup
\bigletter{g}
\item\relax\sphinxstyleindexentry{generate\_genotype\_corpus}\sphinxstyleindexpageref{generate_genotype_corpus:\detokenize{module-generate_genotype_corpus}}
\indexspace
\bigletter{m}
\item\relax\sphinxstyleindexentry{merge\_genotype\_corpora}\sphinxstyleindexpageref{merge_genotype_corpora:\detokenize{module-merge_genotype_corpora}}
\indexspace
\bigletter{s}
\item\relax\sphinxstyleindexentry{simulate\_data}\sphinxstyleindexpageref{simulate_data:\detokenize{module-simulate_data}}
\indexspace
\bigletter{t}
\item\relax\sphinxstyleindexentry{test\_runtime}\sphinxstyleindexpageref{test_runtime:\detokenize{module-test_runtime}}
\indexspace
\bigletter{u}
\item\relax\sphinxstyleindexentry{utils.argparse\_checks}\sphinxstyleindexpageref{utils:\detokenize{module-utils.argparse_checks}}
\item\relax\sphinxstyleindexentry{utils.data\_simulator}\sphinxstyleindexpageref{utils:\detokenize{module-utils.data_simulator}}
\item\relax\sphinxstyleindexentry{utils.extensional\_model}\sphinxstyleindexpageref{utils:\detokenize{module-utils.extensional_model}}
\item\relax\sphinxstyleindexentry{utils.genotype\_corpus\_generator}\sphinxstyleindexpageref{utils:\detokenize{module-utils.genotype_corpus_generator}}
\item\relax\sphinxstyleindexentry{utils.genotype\_corpus\_merger}\sphinxstyleindexpageref{utils:\detokenize{module-utils.genotype_corpus_merger}}
\item\relax\sphinxstyleindexentry{utils.parametrized\_model}\sphinxstyleindexpageref{utils:\detokenize{module-utils.parametrized_model}}
\item\relax\sphinxstyleindexentry{utils.validation\_utils}\sphinxstyleindexpageref{utils:\detokenize{module-utils.validation_utils}}
\indexspace
\bigletter{v}
\item\relax\sphinxstyleindexentry{validate\_simulated\_data}\sphinxstyleindexpageref{validate_simulated_data:\detokenize{module-validate_simulated_data}}
\end{sphinxtheindex}

\renewcommand{\indexname}{Index}
\footnotesize\raggedright\printindex
\end{document}