
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Getting Started with EpiGEN &#8212; EpiGEN  documentation</title>
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="The script simulate_data.py" href="simulate_data.html" />
    <link rel="prev" title="EpiGEN: an epistasis simulation pipeline" href="index.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="simulate_data.html" title="The script simulate_data.py"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="index.html" title="EpiGEN: an epistasis simulation pipeline"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">EpiGEN  documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="getting-started-with-epigen">
<h1>Getting Started with EpiGEN<a class="headerlink" href="#getting-started-with-epigen" title="Permalink to this headline">¶</a></h1>
<div class="section" id="scope">
<h2>Scope<a class="headerlink" href="#scope" title="Permalink to this headline">¶</a></h2>
<p>EpiGEN is an easy-to-use epistasis simulation pipeline written in Python. It supports epistasis models of arbitrary size, which can be specified either extensionally or via parametrized risk models. Moreover, the user can specify the minor allele frequencies (MAFs) of both noise and disease SNPs, and provide a bias target distribution for the generated phenotypes to simulate observation bias.</p>
</div>
<div class="section" id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h2>
<p>EpiGen is freely available on <a class="reference external" href="https://github.com/baumbachlab/epigen">GitHub</a>. To install it on your machine, simply execute the following line in a terminal:</p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>git clone https://github.com/baumbachlab/epigen
</pre></div>
</div>
</div>
<div class="section" id="usage">
<h2>Usage<a class="headerlink" href="#usage" title="Permalink to this headline">¶</a></h2>
<p>The user interface of EpiGEN consists of three scripts:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">simulate_data.py</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">generate_genotype_corpus.py</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">merge_genotype_corpora.py</span></code></p></li>
</ul>
<p>The script <code class="docutils literal notranslate"><span class="pre">simulate_data.py</span></code> simulates epistasis data on top of a pre-computed genotype corpus. For each chromosome <code class="docutils literal notranslate"><span class="pre">&lt;CHROM&gt;</span></code> and each HAPMAP3 population code <code class="docutils literal notranslate"><span class="pre">&lt;POP&gt;</span></code>, EpiGEN contains a pre-computed corpus for 10000 individuals, which is identified by the prefix <code class="docutils literal notranslate"><span class="pre">corpora/&lt;CHROM&gt;_&lt;POP&gt;_</span></code>. For example, if you want to generate epistasis data with ID 0 for 7500 individuals and 10000 SNPs on top of the pre-computed corpus <code class="docutils literal notranslate"><span class="pre">corpora/1_ASW_</span></code>, where the parametrized epistasis model <code class="docutils literal notranslate"><span class="pre">models/param_model.xml</span></code> acts upon the SNPs with IDs 156, 3, and 1076 in the corpus, you can use <code class="docutils literal notranslate"><span class="pre">simulate_data.py</span></code> as follows:</p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>python3 simulate_data.py --sim-ids <span class="m">0</span> --corpus-id <span class="m">1</span> --pop ASW --inds <span class="m">7500</span> --snps <span class="m">10000</span> --disease-snps <span class="m">156</span> <span class="m">3</span> <span class="m">1076</span> --model models/param_model.xml  
</pre></div>
</div>
<p>As you will notice when executing this command, a large fraction of the runtime of <code class="docutils literal notranslate"><span class="pre">simulate_data.py</span></code> is used for loading the corpora. If you want to simulate data for only a small number of individuals, it is therefore advisable to first compute your own, smaller corpora. You can also speed-up the script by unzipping the corpora before running it.</p>
<p>If you want to use custom corpora instead of the pre-computed ones, you can generate them via the script <code class="docutils literal notranslate"><span class="pre">generate_genotype_corpus.py</span></code>. For example, the corpus <code class="docutils literal notranslate"><span class="pre">corpora/1_ASW_</span></code> shipped with EpiGEN was generated as follows:</p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>python3 generate_genotype_corpus.py --corpus-id <span class="m">1</span> --pop ASW --inds <span class="m">10000</span> --chroms <span class="m">1</span> --compress 
</pre></div>
</div>
<p>Finally, the script <code class="docutils literal notranslate"><span class="pre">merge_genotype_corpora.py</span></code> allows you to merge pre-computed corpora into a larger copus. For instance, the following command merges the pre-computed corpora <code class="docutils literal notranslate"><span class="pre">corpora/1_ASW_</span></code> and <code class="docutils literal notranslate"><span class="pre">corpora/2_ASW_</span></code> into a newly generated corpus <code class="docutils literal notranslate"><span class="pre">corpora/23_ASW_</span></code>:</p>
<div class="highlight-sh notranslate"><div class="highlight"><pre><span></span>python3 merge_genotype_corpora.py --corpus-ids <span class="m">1</span> <span class="m">2</span> --pops ASW ASW --corpus-id <span class="m">23</span> --append SNPS
</pre></div>
</div>
<p>Detailed descriptions of how to use the scripts can be found in the HTML and PDF documentations contained in <code class="docutils literal notranslate"><span class="pre">docs/build/html</span></code> and <code class="docutils literal notranslate"><span class="pre">docs/build/latex</span></code>.</p>
</div>
<div class="section" id="implementing-custom-interaction-models">
<h2>Implementing Custom Interaction Models<a class="headerlink" href="#implementing-custom-interaction-models" title="Permalink to this headline">¶</a></h2>
<p>EpiGEN natively supports four parametrized interaction models: exponential, multiplicative, joint-dominant, and joint-recessive interaction. Further interaction models can easily be implemented by the user. Assume, for instance, that the user wants to implement xor-dominant interaction, i.e., a parametrized interaction model where there is an effect if and only if there is at least one minor allele at exactly one of the SNPs involved in the interaction. Then it suffices to insert the following five lines of code at line 242 of <code class="docutils literal notranslate"><span class="pre">utils/parametrized_model.py</span></code>:</p>
<div class="highlight-py notranslate"><div class="highlight"><pre><span></span><span class="k">elif</span> <span class="n">model_type</span> <span class="o">==</span> <span class="s2">&quot;xor-dominant&quot;</span><span class="p">:</span>
	<span class="k">if</span> <span class="n">np</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">gen_at_snp_set</span><span class="p">[</span><span class="n">poss</span><span class="p">])</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
		<span class="k">return</span> <span class="n">alpha</span>
	<span class="k">else</span><span class="p">:</span>
		<span class="k">return</span> <span class="mi">1</span>
</pre></div>
</div>
<p>For consistency, it is also recommendable to add the string <code class="docutils literal notranslate"><span class="pre">&quot;xor-dominant&quot;</span></code> to the error message on line 249 of <code class="docutils literal notranslate"><span class="pre">utils/parametrized_model.py</span></code>, as well to the list of acceptable interaction types on line 42 of the document type definition <code class="docutils literal notranslate"><span class="pre">models/ParametrizedModel.dtd</span></code>.</p>
</div>
<div class="section" id="requirements">
<h2>Requirements<a class="headerlink" href="#requirements" title="Permalink to this headline">¶</a></h2>
<p>EpiGEN has the following dependencies:</p>
<ul class="simple">
<li><p>Python 3.3 or higher.</p></li>
<li><p>Numpy 1.17.3 or higher.</p></li>
<li><p>Scipy 1.3.1 or higher.</p></li>
<li><p>Matplotlib 3.1.1 or higher.</p></li>
</ul>
<p>Moreover, due to its HAPGEN2 dependency, the script <code class="docutils literal notranslate"><span class="pre">generate_genotype_corpus.py</span></code> needs to be run on a Linux machine or on a machine running macOS 10.14 or lower. However, you can avoid running <code class="docutils literal notranslate"><span class="pre">generate_genotype_corpus.py</span></code> by using the pre-computed corpora and merging them, if necessary.</p>
<p>If you want to re-compile the documentation contained in the <code class="docutils literal notranslate"><span class="pre">docs</span></code> directory, you additionally need to install Sphinx, the extension recommonmark, and the package mock. If you have these packages installed, the HTML and PDF documentations can be re-compiled by executing <code class="docutils literal notranslate"><span class="pre">make</span> <span class="pre">html</span></code> and <code class="docutils literal notranslate"><span class="pre">make</span> <span class="pre">latexpdf</span></code> from the <code class="docutils literal notranslate"><span class="pre">docs</span></code> directory.</p>
</div>
<div class="section" id="license">
<h2>License<a class="headerlink" href="#license" title="Permalink to this headline">¶</a></h2>
<p>All of EpiGEN’s Python sources are licensed under the <a class="reference external" href="https://www.gnu.org/licenses/gpl-3.0.de.html">GNU General Public License 3</a>. However, this license does not cover the HAPGEN2 binaries, which are distributed with EpiGEN and are called by the script <code class="docutils literal notranslate"><span class="pre">generate_genotype_corpus.py</span></code>. HAPGEN2 is property of the University of Oxford and may only be freely used for academic research and in accordance with the license found at <a class="reference external" href="https://mathgen.stats.ox.ac.uk/genetics_software/hapgen/LICENCE">https://mathgen.stats.ox.ac.uk/genetics_software/hapgen/LICENCE</a>. Copies of the GNU General Public License 3 and of the license for HAPGEN2 are distributed with EpiGEN.</p>
</div>
<div class="section" id="citing-epigen">
<h2>Citing EpiGEN<a class="headerlink" href="#citing-epigen" title="Permalink to this headline">¶</a></h2>
<p>If you use EpiGEN, please cite the following paper:</p>
<ul class="simple">
<li><p>D. B. Blumenthal, L. Viola, M. List, J. Baumbach, P. Tieri, T. Kacprowski. “EpiGEN: an epistasis simulation pipeline”. Submitted.</p></li>
</ul>
</div>
<div class="section" id="structure-of-the-repository">
<h2>Structure of the Repository<a class="headerlink" href="#structure-of-the-repository" title="Permalink to this headline">¶</a></h2>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>.
├── README.md                        // README
├── LICENSE                          // A copy of the GNU General Public License 3
├── requirements.txt                 // Lists dependencies
├── simulate_data.py                 // Script to simulate epistasis data
├── generate_genotype_corpus.py      // Script to generate genotype corpus
├── merge_genotype_corpora.py        // Script to merge genotype corpora
├── test_runtime.py                  // Script to test EpiGEN&#39;s runtime performance
├── docs                             // Contains Sphinx documentation
├── sim                              // Output directory for simulated data
├── corpora                          // Output directory for genotype corpora
├── temp                             // Contains auxiliary files 
├── ext                              // Contains external libraries and data
│   ├── HAPGEN2                      // Contains HAPGEN2 binaries and license
│   └── HAPMAP3                      // Contains HAPMAP3 data
├── models                           // Contains epistasis models
│   ├── ParametrizedModel.dtd        // Doctype definition for parametrized models
│   ├── ext_model.ini                // An example of an extensional model
│   ├── param_model.xml              // An example of a parametrized model
│   └── ...                          // Further models
└── utils                            // Contains the core of EpiGEN
    ├── __init__.py                  // __init__ file
    ├── data_simulator.py            // Implements simulation of epistasis data
    ├── genotype_corpus_generator.py // Implements generation of genotype corpora
    ├── genotype_corpusmerger.py     // Implements merging of genotype corpora
    ├── parametrized_model.py.       // Implements parametrized models 
    ├── extensional_model.py.        // Implements extensional models
    └── argparse_checks.py.          // Implements argparse checks
</pre></div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Getting Started with EpiGEN</a><ul>
<li><a class="reference internal" href="#scope">Scope</a></li>
<li><a class="reference internal" href="#installation">Installation</a></li>
<li><a class="reference internal" href="#usage">Usage</a></li>
<li><a class="reference internal" href="#implementing-custom-interaction-models">Implementing Custom Interaction Models</a></li>
<li><a class="reference internal" href="#requirements">Requirements</a></li>
<li><a class="reference internal" href="#license">License</a></li>
<li><a class="reference internal" href="#citing-epigen">Citing EpiGEN</a></li>
<li><a class="reference internal" href="#structure-of-the-repository">Structure of the Repository</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="index.html"
                        title="previous chapter">EpiGEN: an epistasis simulation pipeline</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="simulate_data.html"
                        title="next chapter">The script <code class="docutils literal notranslate"><span class="pre">simulate_data.py</span></code></a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/README.md.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="simulate_data.html" title="The script simulate_data.py"
             >next</a> |</li>
        <li class="right" >
          <a href="index.html" title="EpiGEN: an epistasis simulation pipeline"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">EpiGEN  documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2019, David B. Blumenthal, Lorenzo Viola, Markus List, Jan Baumbach, Tim Kacprowski, Paolo Tieri.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.2.1.
    </div>
  </body>
</html>